---
title: "cbio.tas.103"
author: "Analyst: mcm"
output:
  html_document:
    code_folding: hide
    collapsed: yes
    highlight: tango
    number_sections: no
    theme: yeti
    toc: yes
    toc_float: yes
fontsize: 25pt
---

[secundo report](https://webfs/n/core/Bioinformatics/secondary/Workman/tas/MOLNG-2344/analysis_rnaseq.html)

Setting up R, reading in file with sample info, setting colors for plotting.

``` {r setup, echo=TRUE, message=FALSE, results="hide", warning=FALSE, eval=TRUE}
require(knitr)
opts_knit$set(root.dir="/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data")
library(rmarkdown)
library(RColorBrewer)
```

softlink files from secundo
```{r softlink, engine='bash',eval=FALSE}
ln -s /n/facility/Bioinformatics/secondary/Workman/tas/MOLNG-2344/secundo/bam_stats.tsv bam_stats.tsv
ln -s /n/facility/Bioinformatics/secondary/Workman/tas/MOLNG-2344/secundo/htseq_count.csv htseq_count.csv
ln -s /n/facility/Bioinformatics/secondary/Workman/tas/MOLNG-2344/secundo/ercc_count.csv ercc_count.csv
ln -s /n/facility/Bioinformatics/secondary/Workman/tas/MOLNG-2344/targets.tsv ../code/targets.tsv
mkdir bw
find  /n/facility/Bioinformatics/secondary/Workman/tas/MOLNG-2344/ -name "*.bw" | parallel -j 1 "ln -s {} bw"

#upload track files to tracks server
scp bw/*.bw mcm@tracks:/r/compbio/cbio.tas.103
```

Read in htseq counts generated by secundo from both projects.
```{r read in stuff,echo=TRUE,message=F,results="hide",warning=FALSE,eval=TRUE}
targets_initial<-read.table("../code/targets.tsv",sep="\t",header=T,as.is=T,quote="",comment.char="");
targets<-read.table("../code/targets.txt",sep='\t',header=T,as.is=T,quote="",comment.char="")
colnames(targets)[3]<-"alias"

cols<-c(rep("black",times=3),rep(c(brewer.pal(12,"Set3")[c(4,5,6,7)]),each=3),rep(c(brewer.pal(8,"Dark2"),brewer.pal(9,"Set1")[-6],brewer.pal(8,"Set2")[-4],brewer.pal(12,"Set3")[c(3,6,10,11)]),each=2))
count_df_101<-read.table("../../cbio.tas.101/data/htseq_count.csv",sep=',',as.is=T,header=T)
count_df_102<-read.table("../../cbio.tas.102/data/htseq_count.csv",sep=',',as.is=T,header=T)
count_df<-read.table("htseq_count.csv",sep=',',as.is=T,header=T)
in_all<-intersect(count_df[,1],intersect(count_df_101[,1],count_df_102[,1]))
new_count_df<-data.frame(in_all,count_df_102[match(in_all,count_df_102[,1]),c(7,12:14)],count_df_101[match(in_all,count_df_101[,1]),c(2:6,8:10,14:16)],count_df[match(in_all,count_df[,1]),2:length(count_df[1,])])
new_count_df<-new_count_df[,match(targets$alias,colnames(new_count_df))]
rownames(new_count_df)<-in_all
stuff<-new_count_df[grep("^_",rownames(new_count_df)),]
count_df<-new_count_df[grep("^_",rownames(new_count_df),invert=T),]

expected<-read.table("/n/ngs/tools/pilluminati/spikein_analysis/data/expected.txt",sep='\t',as.is=T,header=T)
ercc_101<-read.table("../../cbio.tas.101/data/ercc_count.csv",sep=',',header=T)
ercc<-read.table("ercc_count.csv",sep=',',header=T)
idxstats<-read.table("/n/data1/genomes/indexes/ERCC/ERCC92.fa.fai")
idxstats_lengths<-idxstats[1:92,2]
idxstats_df<-ercc[2:93,2:length(ercc[1,])]
rownames(idxstats_df)<-idxstats[1:92,1]
```

Make plots of the ercc controls relative to their expected counts.
```{r make ercc pdf,echo=TRUE,message=F,results="hide",warning=FALSE,eval=TRUE}
#pdf("ercc.pdf")
#for(i in 1:length(targets$alias[11:length(targets$alias)]))
#{
#	plot(log2(expected[match(rownames(idxstats_df),expected[,2]),4]),log2(idxstats_df[,i]),ylab="log2 observed count",xlab="log2 expected",ylim=c(-5,16),xlim=c(-5,16),main=targets$alias[i])
#	abline(a=0,b=1,col="blue",lty=2)
#}
dev.off()
```

[ercc.pdf](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/ercc.pdf)

Heatmap of ercc counts.
```{r ercc heatmap,echo=T,message=F,results="hide",warning=F,fig.show="hide",eval=T}
rpm<-t(apply(idxstats_df,1,function(x) x/colSums(idxstats_df)*10^6))
temp<-rpm
temp2<-rpm
rownames(temp)<-rownames(idxstats_df[,1])
for(i in 1:length(idxstats_df[,1]))
{
	temp2[i,]<-temp[i,]/idxstats_lengths[i]*1000
}
idxstats_rpkm<-temp2
source("/n/projects/mcm/R/utilities/mcm_image.R")
idxstats_rpkm_sub<-idxstats_rpkm[apply(idxstats_rpkm,1,max,na.rm=T)>100,]
h<-heatmap(as.matrix(idxstats_rpkm_sub))
pdf("heatmap_ercc_rpkm.pdf")
source("/n/projects/mcm/R/utilities/rownorm.R")
temp<-row.norm(log2(idxstats_rpkm_sub),divSD=T)
finite.iv<-apply(temp,2,is.finite)
temp[!finite.iv]<- -3
temp[temp < -2] <- -2
temp[temp > 2 ] <- 2
nf<-layout(matrix(data=c(1,1,1,2,0,0),nrow=3,ncol=2),widths=c(13,3),TRUE)
mcm_image(temp[rev(h$rowInd),h$colInd],colors=colorRampPalette(brewer.pal(9,"Reds"))(100),zlim=c(-2,2),cexRow=1,lab=5,title="ERCC controls log2(rpkms)",cexCol=1)
dev.off()
```
[heatmap_ercc_rpkm.pdf](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/heatmap_ercc_rpkm.pdf)


Plots of bam file statistics
```{r bam stats, echo=TRUE, results="hide", warning=TRUE, message=FALSE}
bam_stats_101<-read.table("../../cbio.tas.101/data/bam_stats.tsv",sep='\t',as.is=T,header=T)
bam_stats_102<-read.table("../../cbio.tas.102/data/bam_stats.tsv",sep='\t',as.is=T,header=T)
bam_stats<-read.table("bam_stats.tsv",sep='\t',as.is=T,header=T)
bam_stats_all<-rbind(bam_stats_102[c(6,11:13),],bam_stats_101[c(1:9,13:15),],bam_stats)
bam_stats<-bam_stats_all[match(targets$alias,bam_stats_all[,1]),]

for(i in c(2,4:length(bam_stats[1,])))
{
   png(paste(colnames(bam_stats)[i],".png",sep=''),width=800,height=800,pointsize=16)
   par(mar=c(11,7,3,1))
   barplot(as.numeric(bam_stats[,i]),names.arg=bam_stats[,1],las=2,cex.names=.7,main=colnames(bam_stats)[i],col=cols)
   dev.off()
}
```

Make track headers for secundo generated bigWigs

```{r bw headers, echo=TRUE, results="hide", warning=TRUE, message=FALSE,eval=T}
sink("track_headers.txt")
trackcols<-apply(t(col2rgb(cols)),1,paste,collapse=',')
for(i in 1:length(targets[,1]))
{
   for(j in c("pos","neg"))
   {
      cat(paste("track type=bigWig name=",targets$alias[i],".",j,".rpm.bw graphType=bar visibility=full bigDataUrl=http://tracks.stowers.org/microarray/repo/compbio/cbio.tas.103/",targets$alias[i],".",j,".rpm.bw color=",trackcols[i]," maxHeightPixels=40:40:40",sep=''),"\n");
   }
}
sink()
```

## UCSC Session

[UCSC session](https://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOtherUser=submit&hgS_otherUserName=Mcm&hgS_otherUserSessionName=cbio.tas.103)


```{r edger analysis, echo=TRUE, results="hide",warning=TRUE,message=FALSE,eval=T}
#read in file to get uxon len
f<-read.table("/n/data1/genomes/indexes/hg38/annotation/Ens_91/beds/hg38.Ens_91.gene_uxons.bed",quote='',header=F,sep='\t',as.is=T)
f$len<-f[,3]-f[,2]
f$gene<-sapply(strsplit(f[,4], ":"), "[", 1)
sumlen<-tapply(f$len,f$gene,sum)
lengths<-data.frame(ensid=in_all,length=as.numeric(sumlen[match(in_all,names(sumlen))]))

targets$group<-c("s_cont", "s_cont", "s_cont", "s_kda", "s_kda", "s_kda", "s_kdb", "s_kdb", "s_kdb", "s_kdc", "s_kdc", "s_kdc", "s_kde", "s_kde", "s_kde", "s_contmo", "s_contmo", "s_contpo", "s_contpo", "s_kdamo", "s_kdamo", "s_kdemo", "s_kdemo", "s_kdfmo", "s_kdfmo", "s_kdgmo", "s_kdgmo", "s_kdapo", "s_kdapo", "s_kdepo", "s_kdepo", "s_kdfpo", "s_kdfpo", "s_kdgpo", "s_kdgpo", "s_kdfin", "s_kdfin", "s_kdgin", "s_kdgin", "s_kdtas2mo", "s_kdtas2mo", "s_kdtas4mo", "s_kdtas4mo", "s_kdtas6mo", "s_kdtas6mo", "s_kdtas10mo", "s_kdtas10mo", "s_kdtas11mo", "s_kdtas11mo", "s_kdtas2po", "s_kdtas2po", "s_kdtas4po", "s_kdtas4po", "s_kdtas6po", "s_kdtas6po", "s_kdtas10_11_po", "s_kdtas10_11_po", "s_kdtas10_11_po", "s_kdtas10_11_po", "s_kdtas2in", "s_kdtas2in", "s_kdtas4in", "s_kdtas4in", "s_kdtas6in", "s_kdtas6in", "s_kdtas10_11_in", "s_kdtas10_11_in", "s_kdtas10_11_in", "s_kdtas10_11_in")


#all0.iv<-rowSums(count_df,na.rm=T)==0
#count_df<-count_df[!all0.iv,]
#count_df[is.na(count_df)]<-0
normCounts<-data.frame(matrix(NA,nrow=length(count_df[,1]),ncol=length(targets$group)))
edger.rpkm<-data.frame(matrix(NA,nrow=length(count_df[,1]),ncol=length(targets$group)))
		
library(edgeR)
y <- DGEList(counts=count_df,group=targets$group)
keep <- rowSums(cpm(y)>4) >= 2
y <- y[keep, , keep.lib.sizes=FALSE]
y <- calcNormFactors(y)
y <- estimateCommonDisp(y)
y <- estimateTagwiseDisp(y)
edger.rpkm <- rpkm(y, gene.length=lengths[match(rownames(y$counts),lengths[,1]),2],normalized.lib.sizes=TRUE)
colnames(edger.rpkm)<-paste("rpkm.",targets$alias,sep='')

res<-list()
res[[1]]<-exactTest(y,pair=c("s_cont","s_contpo"))
res[[2]]<-exactTest(y,pair=c("s_contpo","s_kdapo"))
res[[3]]<-exactTest(y,pair=c("s_contpo","s_kdepo"))
res[[4]]<-exactTest(y,pair=c("s_contpo","s_kdfpo"))
res[[5]]<-exactTest(y,pair=c("s_contpo","s_kdgpo"))
res[[6]]<-exactTest(y,pair=c("s_kdtas10_11_in","s_kdtas10_11_po"))
res[[7]]<-exactTest(y,pair=c("s_kdtas10_11_po","s_kdtas2po"))
res[[8]]<-exactTest(y,pair=c("s_kdtas10_11_po","s_kdtas4po"))
res[[9]]<-exactTest(y,pair=c("s_kdtas10_11_po","s_kdtas6po"))
res[[10]]<-exactTest(y,pair=c("s_cont","s_kdamo"))
res[[11]]<-exactTest(y,pair=c("s_cont","s_kdemo"))
res[[12]]<-exactTest(y,pair=c("s_cont","s_kdfin"))
res[[13]]<-exactTest(y,pair=c("s_cont","s_kdgin"))
res[[14]]<-exactTest(y,pair=c("s_cont","s_kdtas10_11_in"))
res[[15]]<-exactTest(y,pair=c("s_kdtas10_11_in","s_kdtas2in"))
res[[16]]<-exactTest(y,pair=c("s_kdtas10_11_in","s_kdtas4in"))
res[[17]]<-exactTest(y,pair=c("s_kdtas10_11_in","s_kdtas6in"))
res[[18]]<-exactTest(y,pair=c("s_cont","s_kda"))
res[[19]]<-exactTest(y,pair=c("s_cont","s_kde"))

conts<-c("contpo_vs_cont","kdapo_vs_contpo","kdepo_vs_contpo","kdfpo_vs_contpo","kdgpo_vs_contpo","kdtas10_11_po_vs_kdtas10_11_in","kdtas2po_vs_kdtas10_11_po","kdtas4po_vs_kdtas10_11_po","kdtas6po_vs_kdtas10_11_po","kdamo_vs_cont","kdemo_vs_cont","kdfin_vs_cont","kdgin_vs_cont","kdtas10_11_in_vs_cont","kdtas2in_vs_kdtas10_11_in","kdtas4in_vs_kdtas10_11_in","kdtas6in_vs_kdtas10_11_in","kda_vs_cont","kde_vs_cont")
for(i in 1:length(conts))
{
	res[[i]]$table$padj<-p.adjust(res[[i]]$table$PValue,method="BH")
}
```

Correlation matrix, log2 normalized counts.
```{r correlation matrix, echo=TRUE, results="hide",warning=TRUE,message=FALSE}
source("/n/projects/mcm/R/utilities/corplot.R")
png("corplot.png",width=1200,height=1000,pointsize=16)
normCounts<-cpm(y, normalized.lib.sizes=TRUE)
log2normCounts<-log2(normCounts)
colnames(log2normCounts)<-targets$alias
rownames(log2normCounts)<-rownames(y$count)
d<-corplotd(log2normCounts,names=targets$alias)
dev.off()
```
![](../data/corplot.png)

## Comparing to old experiment

How many aligned reads in previous experiment vs. this one?

```{r bam_stats, echo=T,results="hide",warning=T,message=F}
targets_prev<-read.table("../../cbio.tas.101/code/targets.tsv",sep='\t',header=T,as.is=T)
bam_stats_prev<-read.table("../../cbio.tas.101/data/bam_stats.tsv",sep='\t',header=T,as.is=T)

png("barplot_old_new.png",height=600,width=600,pointsize=16)
barplot(c(bam_stats_prev$total_alignments,bam_stats$total_alignments),names.arg=c(bam_stats_prev$sample_name,bam_stats$sample_name),cex.names=.7,las=2,ylab="Total alignments",main="# of aligned reads for cbio.tas.101, cbio.tas.103")
mean(bam_stats_prev$total_alignments)
mean(bam_stats$total_alignments)
dev.off()
```
![](../data/barplot_old_new.png)


```{r track headers, echo=T,results="hide",warning=T,message=F}
sink("track_headers.txt")
trackcols<-apply(t(col2rgb(cols)),1,paste,collapse=',')
for(i in 1:length(targets[,1]))
{
   for(j in c("pos","neg"))
   {
      cat(paste("track type=bigWig name=",targets$alias[i],".",j,".rpm.bw graphType=bar visibility=full bigDataUrl=http://tracks.stowers.org/microarray/repo/compbio/cbio.tas.103/",targets$alias[i],".",j,".rpm.bw color=",trackcols[i]," maxHeightPixels=80:80:80",sep=''),"\n");
   }
}
sink()
```

[UCSC session](http://genome.ucsc.edu/cgi-bin/hgTracks?hgS_doOtherUser=submit&hgS_otherUserName=Mcm&hgS_otherUserSessionName=cbio.tas.103)


```{r making summary table, echo=TRUE, results="hide",warning=TRUE,message=FALSE}
summarytable<-data.frame(rownames(y$count),round(edger.rpkm,3),res[[1]]$table,res[[2]]$table,res[[3]]$table,res[[4]]$table,res[[5]]$table,res[[6]]$table,res[[7]]$table,res[[8]]$table,res[[9]]$table,res[[10]]$table,res[[11]]$table,res[[12]]$table,res[[13]]$table,res[[14]]$table,res[[15]]$table,res[[16]]$table,res[[17]]$table,res[[18]]$table,res[[19]]$table)

colnames(summarytable)[1]<-"ensid"
colnames(summarytable)[71:length(summarytable[1,])]<-paste(c(paste(rep(conts,each=4),rep(c(".log2fc",".log2cpm",".pval",".padj"),times=1),sep='')))

#genedata<-read.table("/n/data1/genomes/bowtie-index/hg38/annotation/Ens_91/tables/hg38.Ens_91.gene_data.txt",sep='\t',header=T,quote="",comment.char="")
genedata<-read.table("/n/data1/genomes/indexes/hg38/annotation/Ens_91/tables/hg38.Ens_91.gene_data.txt",sep='\t',header=T,quote="",comment.char="")

summarytable_ann<-data.frame(ensid=summarytable[,1],genedata[match(summarytable[,1],genedata[,1]),c("Name","Chrom", "Start", "End", "Strand", "Biotype","Description")],summarytable[,2:length(summarytable[1,])])
rownames(summarytable_ann)<-summarytable_ann$ensid
summarytable_ann.sort<-summarytable_ann[order(summarytable_ann[,paste(conts[2],".padj",sep='')]),]
write.table(summarytable_ann.sort,"summary_genes.tsv",sep='\t',quote=F,row.names=F)
```

Here is a summary file with all genes and all contrasts.

[summary_genes.tsv](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/summary_genes.tsv)

```{r writing out DE gene files, echo=TRUE, results="hide",warning=TRUE,message=FALSE}
genes_up<-list()
genes_dn<-list()
genes_de<-list()
for(i in 1:length(conts))
{
	iv.de<-summarytable_ann.sort[,paste(conts[i],".padj",sep='')] < .05 & abs(summarytable_ann.sort[,paste(conts[i],".log2fc",sep='')]) > 1 
	genes_de[[i]]<-summarytable_ann.sort[iv.de,c("ensid",paste(conts[i],".log2fc",sep=''))]
	iv.up<-summarytable_ann.sort[,paste(conts[i],".padj",sep='')] < .05 & summarytable_ann.sort[,paste(conts[i],".log2fc",sep='')] > 1 
	genes_up[[i]]<-summarytable_ann.sort[iv.up,1]
	iv.dn<-summarytable_ann.sort[,paste(conts[i],".padj",sep='')] < .05 & summarytable_ann.sort[,paste(conts[i],".log2fc",sep='')] < -1 
	genes_dn[[i]]<-summarytable_ann.sort[iv.dn,1]
	#write.table(summarytable_ann.sort[iv.up,],paste("../results/DE/",conts[i],".de.up.xls",sep=''),sep='\t',quote=F,row.names=F)
	#write.table(summarytable_ann.sort[iv.dn,],paste("../results/DE/",conts[i],".de.dn.xls",sep=''),sep='\t',quote=F,row.names=F)
}
names(genes_up)<-conts
names(genes_dn)<-conts
names(genes_de)<-conts
```

Here are files for each contrast:

[contpo_vs_cont.de.up.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/contpo_vs_cont.de.up.xls)  
[contpo_vs_cont.de.dn.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/contpo_vs_cont.de.dn.xls)  

[kdapo_vs_contpo.de.up.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdapo_vs_kdamo.de.up.xls)  
[kdapo_vs_contpo.de.dn.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdapo_vs_kdamo.de.dn.xls)  

[kdepo_vs_contpo.de.up.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdepo_vs_kdemo.de.up.xls)  
[kdepo_vs_contpo.de.dn.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdepo_vs_kdemo.de.dn.xls)  

[kdfpo_vs_contpo.de.up.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdfpo_vs_kdfin.de.up.xls)  
[kdfpo_vs_contpo.de.dn.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdfpo_vs_kdfin.de.dn.xls)  

[kdgpo_vs_contpo.de.up.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdgpo_vs_kdgin.de.up.xls)  
[kdgpo_vs_contpo.de.dn.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdgpo_vs_kdgin.de.dn.xls)  

[kdtas10_11_po_vs_kdtas10_11_in.de.up.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdtas10_11_po_vs_kdtas10_11_in.de.up.xls)  
[kdtas10_11_po_vs_kdtas10_11_in.de.dn.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdtas10_11_po_vs_kdtas10_11_in.de.dn.xls)  

[kdtas2po_vs_kdtas10_11_po.de.up.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdtas2po_vs_kdtas10_11_po.de.up.xls)  
[kdtas2po_vs_kdtas10_11_po.de.dn.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdtas2po_vs_kdtas10_11_po.de.dn.xls)  

[kdtas6po_vs_kdtas10_11_po.de.up.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdtas6po_vs_kdtas10_11_po.de.up.xls)  
[kdtas6po_vs_kdtas10_11_po.de.dn.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdtas6po_vs_kdtas10_11_po.de.dn.xls)  

[kdtas4po_vs_kdtas10_11_po.de.up.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdtas4po_vs_kdtas10_11_po.de.up.xls)  
[kdtas4po_vs_kdtas10_11_po.de.dn.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdtas4po_vs_kdtas10_11_po.de.dn.xls)  

[kdamo_vs_cont.de.up.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdamo_vs_cont.de.up.xls)
[kdamo_vs_cont.de.dn.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdamo_vs_cont.de.dn.xls)

[kdemo_vs_cont.de.up.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdemo_vs_cont.de.up.xls)
[kdemo_vs_cont.de.dn.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdemo_vs_cont.de.dn.xls)

[kdfin_vs_cont.de.up.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdfin_vs_cont.de.up.xls)
[kdfin_vs_cont.de.dn.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdfin_vs_cont.de.dn.xls)

[kdgin_vs_cont.de.up.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdgin_vs_cont.de.up.xls)
[kdgin_vs_cont.de.dn.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdgin_vs_cont.de.dn.xls)

[kdtas10_11_in_vs_cont.de.up.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdtas10_11_in_vs_cont.de.up.xls)
[kdtas10_11_in_vs_cont.de.dn.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdtas10_11_in_vs_cont.de.dn.xls)

[kdtas2in_vs_kdtas10_11_in.de.up.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdtas2in_vs_kdtas10_11_in.de.up.xls)
[kdtas2in_vs_kdtas10_11_in.de.dn.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdtas2in_vs_kdtas10_11_in.de.dn.xls)

[kdtas4in_vs_kdtas10_11_in.de.up.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdtas4in_vs_kdtas10_11_in.de.up.xls)
[kdtas4in_vs_kdtas10_11_in.de.dn.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdtas4in_vs_kdtas10_11_in.de.dn.xls)

[kdtas6in_vs_kdtas10_11_in.de.up.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdtas6in_vs_kdtas10_11_in.de.up.xls)
[kdtas6in_vs_kdtas10_11_in.de.dn.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kdtas6in_vs_kdtas10_11_in.de.dn.xls)

[kda_vs_cont.de.up.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kda_vs_cont.de.up.xls)
[kda_vs_cont.de.dn.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kda_vs_cont.de.dn.xls)

[kde_vs_cont.de.up.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kde_vs_cont.de.up.xls)
[kde_vs_cont.de.dn.xls](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE/kde_vs_cont.de.dn.xls)

```{r hist of rpkm, echo=TRUE, results="hide",warning=TRUE, message=FALSE}
png("hist_rpkm.png",height=600,width=600,pointsize=16)
hist(log2(unlist(edger.rpkm)),breaks=100)
dev.off()
```

```{r,out.width="600px"}
	knitr::include_graphics("../data/hist_rpkm.png")
```

Venn diagram - mutant input to control input, mutant polysome to control polysome. We are more interested in genes that change only in mutant polysome, indicating they are translationally affected.

```{r venn ctrl, echo=TRUE, results="hide",warning=TRUE, message=FALSE,eval=FALSE}
#function to write out genes in venn diagram sections
writegenes<-function(genes,filenamepart)
{
	df<-genedata[genedata[,1] %in% genes,c("Gene_ID","Name","Biotype","Description")]
	write.table(df,paste(filenamepart,".",length(df[,1]),".txt",sep=''),row.names=F,quote=F,sep='\t')
}

de_in_kdamo_vs_cont<-genes_de$kdamo_vs_cont[,1]
de_in_kdapo_vs_contpo<-genes_de$kdapo_vs_contpo[,1]

library(Vennerable)
Vcombo<-Venn(SetNames=c("de.kdamo_vs_cont","de.kdapo_vs_contpo"),Sets=list(de_in_kdamo_vs_cont,de_in_kdapo_vs_contpo))
if(!is.null(Vcombo@IntersectionSets$`11`))
{
	png("2venn.kdamo_vs_cont.kdapo_vs_contpo.png",height=600,width=600,pointsize=16)
	plot(Vcombo)
	writegenes(Vcombo@IntersectionSets$`11`,paste("2venn.kdamo_vs_cont.kdapo_vs_contpo.11",sep=''))
	writegenes(Vcombo@IntersectionSets$`01`,paste("2venn.kdamo_vs_cont.kdapo_vs_contpo.01",sep=''))
	writegenes(Vcombo@IntersectionSets$`10`,paste("2venn.kdamo_vs_cont.kdapo_vs_contpo.10",sep=''))
	dev.off()
}	


de_in_kdemo_vs_cont<-genes_de$kdemo_vs_cont
de_in_kdepo_vs_contpo<-genes_de$kdepo_vs_contpo

Vcombo<-Venn(SetNames=c("de.kdemo_vs_cont","de.kdepo_vs_contpo"),Sets=list(de_in_kdemo_vs_cont,de_in_kdepo_vs_contpo))
if(!is.null(Vcombo@IntersectionSets$`11`))
{
	png("2venn.kdemo_vs_cont.kdepo_vs_contpo.png",height=600,width=600,pointsize=16)
	plot(Vcombo)
	writegenes(Vcombo@IntersectionSets$`11`,paste("2venn.kdemo_vs_cont.kdepo_vs_contpo.11",sep=''))
	writegenes(Vcombo@IntersectionSets$`01`,paste("2venn.kdemo_vs_cont.kdepo_vs_contpo.01",sep=''))
	writegenes(Vcombo@IntersectionSets$`10`,paste("2venn.kdemo_vs_cont.kdepo_vs_contpo.10",sep=''))
	dev.off()
}	

de_in_kdfin_vs_cont<-genes_de$kdfin_vs_cont
de_in_kdfpo_vs_contpo<-genes_de$kdfpo_vs_contpo

Vcombo<-Venn(SetNames=c("de.kdfin_vs_cont","de.kdfpo_vs_contpo"),Sets=list(de_in_kdfin_vs_cont,de_in_kdfpo_vs_contpo))
if(!is.null(Vcombo@IntersectionSets$`11`))
{
	png("2venn.kdfin_vs_cont.kdfpo_vs_contpo.png",height=600,width=600,pointsize=16)
	plot(Vcombo)
	writegenes(Vcombo@IntersectionSets$`11`,paste("2venn.kdfin_vs_cont.kdfpo_vs_contpo.11",sep=''))
	writegenes(Vcombo@IntersectionSets$`01`,paste("2venn.kdfin_vs_cont.kdfpo_vs_contpo.01",sep=''))
	writegenes(Vcombo@IntersectionSets$`10`,paste("2venn.kdfin_vs_cont.kdfpo_vs_contpo.10",sep=''))
	dev.off()
}	

de_in_kdgin_vs_cont<-genes_de$kdgin_vs_cont
de_in_kdgpo_vs_contpo<-genes_de$kdgpo_vs_contpo

library(Vennerable)
Vcombo<-Venn(SetNames=c("de.kdgin_vs_cont","de.kdgpo_vs_contpo"),Sets=list(de_in_kdgin_vs_cont,de_in_kdgpo_vs_contpo))
if(!is.null(Vcombo@IntersectionSets$`11`))
{
	png("2venn.kdgin_vs_cont.kdgpo_vs_contpo.png",height=600,width=600,pointsize=16)
	plot(Vcombo)
	writegenes(Vcombo@IntersectionSets$`11`,paste("2venn.kdgin_vs_cont.kdgpo_vs_contpo.11",sep=''))
	writegenes(Vcombo@IntersectionSets$`01`,paste("2venn.kdgin_vs_cont.kdgpo_vs_contpo.01",sep=''))
	writegenes(Vcombo@IntersectionSets$`10`,paste("2venn.kdgin_vs_cont.kdgpo_vs_contpo.10",sep=''))
	dev.off()
}	


de_in_kdtas2in_vs_cont<-genes_de$kdtas2in_vs_cont
de_in_kdtas2po_vs_contpo<-genes_de$kdtas2po_vs_contpo

Vcombo<-Venn(SetNames=c("de.kdtas2in_vs_cont","de.kdtas2po_vs_contpo"),Sets=list(de_in_kdtas2in_vs_cont,de_in_kdtas2po_vs_contpo))
if(!is.null(Vcombo@IntersectionSets$`11`))
{
	png("2venn.kdtas2in_vs_cont.kdtas2po_vs_contpo.png",height=600,width=600,pointsize=16)
	plot(Vcombo)
	writegenes(Vcombo@IntersectionSets$`11`,paste("2venn.kdtas2in_vs_cont.kdtas2po_vs_contpo.11",sep=''))
	writegenes(Vcombo@IntersectionSets$`01`,paste("2venn.kdtas2in_vs_cont.kdtas2po_vs_contpo.01",sep=''))
	writegenes(Vcombo@IntersectionSets$`10`,paste("2venn.kdtas2in_vs_cont.kdtas2po_vs_contpo.10",sep=''))
	dev.off()
}	

de_in_kdtas4in_vs_cont<-genes_de$kdtas4in_vs_cont
de_in_kdtas4po_vs_contpo<-genes_de$kdtas4po_vs_contpo

Vcombo<-Venn(SetNames=c("de.kdtas4in_vs_cont","de.kdtas4po_vs_contpo"),Sets=list(de_in_kdtas4in_vs_cont,de_in_kdtas4po_vs_contpo))
if(!is.null(Vcombo@IntersectionSets$`11`))
{
	png("2venn.kdtas4in_vs_cont.kdtas4po_vs_contpo.png",height=600,width=600,pointsize=16)
	plot(Vcombo)
	writegenes(Vcombo@IntersectionSets$`11`,paste("2venn.kdtas4in_vs_cont.kdtas4po_vs_contpo.11",sep=''))
	writegenes(Vcombo@IntersectionSets$`01`,paste("2venn.kdtas4in_vs_cont.kdtas4po_vs_contpo.01",sep=''))
	writegenes(Vcombo@IntersectionSets$`10`,paste("2venn.kdtas4in_vs_cont.kdtas4po_vs_contpo.10",sep=''))
	dev.off()
}	

de_in_kdtas6in_vs_cont<-genes_de$kdtas6in_vs_cont
de_in_kdtas6po_vs_contpo<-genes_de$kdtas6po_vs_contpo

Vcombo<-Venn(SetNames=c("de.kdtas6in_vs_cont","de.kdtas6po_vs_contpo"),Sets=list(de_in_kdtas6in_vs_cont,de_in_kdtas6po_vs_contpo))
if(!is.null(Vcombo@IntersectionSets$`11`))
{
	png("2venn.kdtas6in_vs_cont.kdtas6po_vs_contpo.png",height=600,width=600,pointsize=16)
	plot(Vcombo)
	writegenes(Vcombo@IntersectionSets$`11`,paste("2venn.kdtas6in_vs_cont.kdtas6po_vs_contpo.11",sep=''))
	writegenes(Vcombo@IntersectionSets$`01`,paste("2venn.kdtas6in_vs_cont.kdtas6po_vs_contpo.01",sep=''))
	writegenes(Vcombo@IntersectionSets$`10`,paste("2venn.kdtas6in_vs_cont.kdtas6po_vs_contpo.10",sep=''))
	dev.off()
}	

de_in_kdtas10in_vs_cont<-genes_de$kdtas10in_vs_cont
de_in_kdtas10po_vs_contpo<-genes_de$kdtas10po_vs_contpo

Vcombo<-Venn(SetNames=c("de.kdtas10in_vs_cont","de.kdtas10po_vs_contpo"),Sets=list(de_in_kdtas10in_vs_cont,de_in_kdtas10po_vs_contpo))
if(!is.null(Vcombo@IntersectionSets$`11`))
{
	png("2venn.kdtas10in_vs_cont.kdtas10po_vs_contpo.png",height=600,width=600,pointsize=16)
	plot(Vcombo)
	writegenes(Vcombo@IntersectionSets$`11`,paste("2venn.kdtas10in_vs_cont.kdtas10po_vs_contpo.11",sep=''))
	writegenes(Vcombo@IntersectionSets$`01`,paste("2venn.kdtas10in_vs_cont.kdtas10po_vs_contpo.01",sep=''))
	writegenes(Vcombo@IntersectionSets$`10`,paste("2venn.kdtas10in_vs_cont.kdtas10po_vs_contpo.10",sep=''))
	dev.off()
}	

de_in_kdtas11in_vs_cont<-genes_de$kdtas11in_vs_cont
de_in_kdtas11po_vs_contpo<-genes_de$kdtas11po_vs_contpo

Vcombo<-Venn(SetNames=c("de.kdtas11in_vs_cont","de.kdtas11po_vs_contpo"),Sets=list(de_in_kdtas11in_vs_cont,de_in_kdtas11po_vs_contpo))
if(!is.null(Vcombo@IntersectionSets$`11`))
{
	png("2venn.kdtas11in_vs_cont.kdtas11po_vs_contpo.png",height=600,width=600,pointsize=16)
	plot(Vcombo)
	writegenes(Vcombo@IntersectionSets$`11`,paste("2venn.kdtas11in_vs_cont.kdtas11po_vs_contpo.11",sep=''))
	writegenes(Vcombo@IntersectionSets$`01`,paste("2venn.kdtas11in_vs_cont.kdtas11po_vs_contpo.01",sep=''))
	writegenes(Vcombo@IntersectionSets$`10`,paste("2venn.kdtas11in_vs_cont.kdtas11po_vs_contpo.10",sep=''))
	dev.off()
}	

de_in_kda_vs_cont<-genes_de$kda_vs_cont[,1]
de_in_kdapo_vs_contpo<-genes_de$kdapo_vs_contpo[,1]

Vcombo<-Venn(SetNames=c("de.kda_vs_cont","de.kdapo_vs_contpo"),Sets=list(de_in_kda_vs_cont,de_in_kdapo_vs_contpo))
	png("2venn.kda_vs_cont.kdapo_vs_contpo.png",height=600,width=600,pointsize=16)
	plot(Vcombo)
	writegenes(Vcombo@IntersectionSets$`11`,paste("2venn.kda_vs_cont.kdapo_vs_contpo.11",sep=''))
	writegenes(Vcombo@IntersectionSets$`01`,paste("2venn.kda_vs_cont.kdapo_vs_contpo.01",sep=''))
	writegenes(Vcombo@IntersectionSets$`10`,paste("2venn.kda_vs_cont.kdapo_vs_contpo.10",sep=''))
	dev.off()

de_in_kde_vs_cont<-genes_de$kde_vs_cont[,1]
de_in_kdepo_vs_contpo<-genes_de$kdepo_vs_contpo[,1]

Vcombo<-Venn(SetNames=c("de.kde_vs_cont","de.kdepo_vs_contpo"),Sets=list(de_in_kde_vs_cont,de_in_kdepo_vs_contpo))
if(!is.null(Vcombo@IntersectionSets$`11`))
{
	png("2venn.kde_vs_cont.kdepo_vs_contpo.png",height=600,width=600,pointsize=16)
	plot(Vcombo)
	writegenes(Vcombo@IntersectionSets$`11`,paste("2venn.kde_vs_cont.kdepo_vs_contpo.11",sep=''))
	writegenes(Vcombo@IntersectionSets$`01`,paste("2venn.kde_vs_cont.kdepo_vs_contpo.01",sep=''))
	writegenes(Vcombo@IntersectionSets$`10`,paste("2venn.kde_vs_cont.kdepo_vs_contpo.10",sep=''))
	dev.off()
}	
```

KDA

```{r,out.width="600px"}
knitr::include_graphics("../data/2venn.kda_vs_cont.kdapo_vs_contpo.png")
```

444 genes from right side:

[2venn.kda_vs_cont.kdapo_vs_contpo.01.444.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/2venn.kda_vs_cont.kdapo_vs_contpo.01.444.txt)

KDAMO

```{r,out.width="600px"}
knitr::include_graphics("../data/2venn.kdamo_vs_cont.kdapo_vs_contpo.png")
```

379 genes from the right side:

[2venn.kdamo_vs_cont.kdapo_vs_contpo.01.379.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/2venn.kdamo_vs_cont.kdapo_vs_contpo.01.379.txt)

KDE

```{r,out.width="600px"}
knitr::include_graphics("../data/2venn.kde_vs_cont.kdepo_vs_contpo.png")
```

703 genes from right side:

[2venn.kde_vs_cont.kdepo_vs_contpo.01.703.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/2venn.kde_vs_cont.kdepo_vs_contpo.01.703.txt)

KDEMO

```{r,out.width="600px"}
knitr::include_graphics("../data/2venn.kdemo_vs_cont.kdepo_vs_contpo.png")
```

641 genes from right side:

[2venn.kdemo_vs_cont.kdepo_vs_contpo.01.641.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/2venn.kdemo_vs_cont.kdepo_vs_contpo.01.641.txt)


KDF

```{r,out.width="600px"}
knitr::include_graphics("../data/2venn.kdfin_vs_cont.kdfpo_vs_contpo.png")
```

5760 genes from right side:

[2venn.kdfin_vs_cont.kdfpo_vs_contpo.01.5760.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/2venn.kdfin_vs_cont.kdfpo_vs_contpo.01.5760.txt)

KDG

```{r,out.width="600px"}
knitr::include_graphics("../data/2venn.kdgin_vs_cont.kdgpo_vs_contpo.png")
```

539 genes from right side:

[2venn.kdgin_vs_cont.kdgpo_vs_contpo.01.539.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/2venn.kdgin_vs_cont.kdgpo_vs_contpo.01.539.txt)

KDTAS2

```{r,out.width="600px"}
knitr::include_graphics("../data/2venn.kdtas2in_vs_cont.kdtas2po_vs_contpo.png")
```

1790 genes from right side:

[2venn.kdtas2in_vs_cont.kdtas2po_vs_contpo.01.1790.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/2venn.kdtas2in_vs_cont.kdtas2po_vs_contpo.01.1790.txt)

KDTAS4

```{r,out.width="600px"}
knitr::include_graphics("../data/2venn.kdtas4in_vs_cont.kdtas4po_vs_contpo.png")
```

1333 genes from right side:

[2venn.kdtas4in_vs_cont.kdtas4po_vs_contpo.01.1333.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/2venn.kdtas4in_vs_cont.kdtas4po_vs_contpo.01.1333.txt)

KDTAS6

```{r,out.width="600px"}
knitr::include_graphics("../data/2venn.kdtas6in_vs_cont.kdtas6po_vs_contpo.png")
```

3714 genes from right side:

[2venn.kdtas6in_vs_cont.kdtas6po_vs_contpo.01.3714.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/2venn.kdtas6in_vs_cont.kdtas6po_vs_contpo.01.3714.txt)

KDTAS10

```{r,out.width="600px"}
knitr::include_graphics("../data/2venn.kdtas10in_vs_cont.kdtas10po_vs_contpo.png")
```

2039 genes from right side:

[2venn.kdtas10in_vs_cont.kdtas10po_vs_contpo.01.2039.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/2venn.kdtas10in_vs_cont.kdtas10po_vs_contpo.01.2039.txt)

KDTAS11

```{r,out.width="600px"}
knitr::include_graphics("../data/2venn.kdtas11in_vs_cont.kdtas11po_vs_contpo.png")
```

3469 genes from right side:

[2venn.kdtas11in_vs_cont.kdtas11po_vs_contpo.01.3469.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/2venn.kdtas11in_vs_cont.kdtas11po_vs_contpo.01.3469.txt)

GO analysis of DE lists and parts of venn diagrams

```{r go, echo=TRUE, results="hide",warning=TRUE, message=FALSE,eval=FALSE}
source("../code/go_analysis.R")
```

[go_barplots.pdf](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/go_barplots.pdf)  
[go_barplots.venn.pdf](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/go_barplots.venn.pdf)

Tamaki also requested to combine the kda/kdg/tas2 lists (union) and search for enriched go terms in BP and MF.

[go_barplot.comb.kda.kdg.tas2.pdf](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/go_barplot.comb.kda.kdg.tas2.pdf)

Combining the lists (intersection)

[go_barplot.int.kda.kdg.tas2.pdf](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/go_barplot.int.kda.kdg.tas2.pdf)



## Overlapping the knockdowns

How do the genes that change in polysomes vs. controlpo intersect?

```{r overlapping knockdowns, echo=TRUE, results="hide",warning=TRUE, message=FALSE,eval=FALSE}
po_only_files<-c("2venn.kda_vs_cont.kdapo_vs_contpo.01.444.txt", "2venn.kde_vs_cont.kdepo_vs_contpo.01.703.txt", "2venn.kdfin_vs_cont.kdfpo_vs_contpo.01.5760.txt", "2venn.kdgin_vs_cont.kdgpo_vs_contpo.01.539.txt")
f<-list()
for(i in 1:length(po_only_files))
{
	f[[i]]<-read.table(po_only_files[i],sep='\t',as.is=T,quote="",comment.char="",header=T)[,1]
}

setnames<-c("kdapo_444","kdepo_703","kdfpo_5760","kdgpo_539")
Vcombo<-Venn(SetNames=setnames,Sets=f)

png("4venn.yellow_venns.chow_ruskey.png",height=600,width=800,pointsize=16)
plot(Vcombo)
dev.off()
png("4venn.yellow_venns.ellipse.png",height=600,width=800,pointsize=16)
plot(Vcombo,type="ellipses")
dev.off()
writegenes(Vcombo@IntersectionSets$`1111`,paste("4venn.yellowvenn.1111",sep=''))
writegenes(Vcombo@IntersectionSets$`1110`,paste("4venn.yellowvenn.1110",sep=''))
writegenes(Vcombo@IntersectionSets$`1010`,paste("4venn.yellowvenn.1010",sep=''))
writegenes(Vcombo@IntersectionSets$`0110`,paste("4venn.yellowvenn.0110",sep=''))
```

Chow-ruskey style venn of all the yellow parts of the venn diagrams (polysome-only de genes) for kda, kde, kdf, kdg.

```{r,out.width="600px"}
knitr::include_graphics("../data/4venn.yellow_venns.chow_ruskey.png")
```

Ellipse style venn

```{r,out.width="600px"}
knitr::include_graphics("../data/4venn.yellow_venns.ellipse.png")
```

There are 172 genes that are present in all the selections:

[4venn.yellowvenn.1111.172.txt](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/4venn.yellowvenn.1111.172.txt)

Venn diagrams of kdtas data - using as a control tas10 and tas11.

First, compare tas10 and tas11.

```{r compare tas10 and tas11, echo=TRUE, results="hide",warning=TRUE, message=FALSE,eval=TRUE}
png("corplot.tas10-11.png",height=600,width=800,pointsize=16)
corplotd(log2normCounts[,c("s_kdtas10po1","s_kdtas10po2","s_kdtas11po1","s_kdtas11po2","s_kdtas10in1","s_kdtas10in2","s_kdtas11in1","s_kdtas11in2")])
dev.off()

png("scatter.tas10-11in.png",height=600,width=600,pointsize=16)
par(mfrow=c(2,2))
plot(log2normCounts[,"s_kdtas10in1"],log2normCounts[,"s_kdtas11in1"],xlab="log2normCount kdtas10in1",ylab="log2normCount kdtas11in1",ylim=c(-5,15),xlim=c(-5,15))
plot(log2normCounts[,"s_kdtas10in2"],log2normCounts[,"s_kdtas11in2"],xlab="log2normCount kdtas10in2",ylab="log2normCount kdtas11in2",ylim=c(-5,15),xlim=c(-5,15))
plot(log2normCounts[,"s_kdtas10in1"],log2normCounts[,"s_kdtas11in2"],xlab="log2normCount kdtas10in1",ylab="log2normCount kdtas11in2",ylim=c(-5,15),xlim=c(-5,15))
plot(log2normCounts[,"s_kdtas10in2"],log2normCounts[,"s_kdtas11in1"],xlab="log2normCount kdtas10in2",ylab="log2normCount kdtas11in1",ylim=c(-5,15),xlim=c(-5,15))
dev.off()

png("scatter.tas10-11po.png",height=600,width=600,pointsize=16)
par(mfrow=c(2,2))
plot(log2normCounts[,"s_kdtas10po1"],log2normCounts[,"s_kdtas11po1"],xlab="log2normCount kdtas10po1",ylab="log2normCount kdtas11po1",ylim=c(-5,15),xlim=c(-5,15))
plot(log2normCounts[,"s_kdtas10po2"],log2normCounts[,"s_kdtas11po2"],xlab="log2normCount kdtas10po2",ylab="log2normCount kdtas11po2",ylim=c(-5,15),xlim=c(-5,15))
plot(log2normCounts[,"s_kdtas10po1"],log2normCounts[,"s_kdtas11po2"],xlab="log2normCount kdtas10po1",ylab="log2normCount kdtas11po2",ylim=c(-5,15),xlim=c(-5,15))
plot(log2normCounts[,"s_kdtas10po2"],log2normCounts[,"s_kdtas11po1"],xlab="log2normCount kdtas10po2",ylab="log2normCount kdtas11po1",ylim=c(-5,15),xlim=c(-5,15))
dev.off()
```

```{r,out.width="600px"}
knitr::include_graphics("../data/corplot.tas10-11.png")
knitr::include_graphics("../data/scatter.tas10-11in.png")
knitr::include_graphics("../data/scatter.tas10-11po.png")
```


```{r venns with tas10 and tas11, echo=TRUE, results="hide",warning=TRUE, message=FALSE,eval=FALSE}
de_in_tas2_vs_tas10_11<-genes_de$kdtas2in_vs_kdtas10_11_in
de_in_tas2po_vs_tas10_11_po<-genes_de$kdtas2po_vs_kdtas10_11_po

Vcombo<-Venn(SetNames=c("de.kdtas2in_vs_kdtas10_11_in","de.kdtas2po_vs_kdtas10_11_po"),Sets=list(de_in_tas2_vs_tas10_11,de_in_tas2po_vs_tas10_11_po))
png("2venn.kdtas2in_vs_kdtas10_11_in.kdtas2po_vs_kdtas10_11_po.png",height=600,width=600,pointsize=16)
plot(Vcombo)
writegenes(Vcombo@IntersectionSets$`11`,paste("2venn.kdtas2in_vs_kdtas10_11_in.kdtas2po_vs_kdtas10_11_po.11",sep=''))
writegenes(Vcombo@IntersectionSets$`01`,paste("2venn.kdtas2in_vs_kdtas10_11_in.kdtas2po_vs_kdtas10_11_po.01",sep=''))
writegenes(Vcombo@IntersectionSets$`10`,paste("2venn.kdtas2in_vs_kdtas10_11_in.kdtas2po_vs_kdtas10_11_po.10",sep=''))
dev.off()

de_in_tas4_vs_tas10_11<-genes_de$kdtas4in_vs_kdtas10_11_in
de_in_tas4po_vs_tas10_11_po<-genes_de$kdtas4po_vs_kdtas10_11_po

Vcombo<-Venn(SetNames=c("de.kdtas4in_vs_kdtas10_11_in","de.kdtas4po_vs_kdtas10_11_po"),Sets=list(de_in_tas4_vs_tas10_11,de_in_tas4po_vs_tas10_11_po))
if(!is.null(Vcombo@IntersectionSets$`11`))
{
	png("2venn.kdtas4in_vs_kdtas10_11_in.kdtas4po_vs_kdtas10_11_po.png",height=600,width=600,pointsize=16)
	plot(Vcombo)
	writegenes(Vcombo@IntersectionSets$`11`,paste("2venn.kdtas4in_vs_kdtas10_11_in.kdtas4po_vs_kdtas10_11_po.11",sep=''))
	writegenes(Vcombo@IntersectionSets$`01`,paste("2venn.kdtas4in_vs_kdtas10_11_in.kdtas4po_vs_kdtas10_11_po.01",sep=''))
	writegenes(Vcombo@IntersectionSets$`10`,paste("2venn.kdtas4in_vs_kdtas10_11_in.kdtas4po_vs_kdtas10_11_po.10",sep=''))
	dev.off()
}	

de_in_tas6_vs_tas10_11<-genes_de$kdtas6in_vs_kdtas10_11_in
de_in_tas6po_vs_tas10_11_po<-genes_de$kdtas6po_vs_kdtas10_11_po

Vcombo<-Venn(SetNames=c("de.kdtas6in_vs_kdtas10_11_in","de.kdtas6po_vs_kdtas10_11_po"),Sets=list(de_in_tas6_vs_tas10_11,de_in_tas6po_vs_tas10_11_po))
if(!is.null(Vcombo@IntersectionSets$`11`))
{
	png("2venn.kdtas6in_vs_kdtas10_11_in.kdtas6po_vs_kdtas10_11_po.png",height=600,width=600,pointsize=16)
	plot(Vcombo)
	writegenes(Vcombo@IntersectionSets$`11`,paste("2venn.kdtas6in_vs_kdtas10_11_in.kdtas6po_vs_kdtas10_11_po.11",sep=''))
	writegenes(Vcombo@IntersectionSets$`01`,paste("2venn.kdtas6in_vs_kdtas10_11_in.kdtas6po_vs_kdtas10_11_po.01",sep=''))
	writegenes(Vcombo@IntersectionSets$`10`,paste("2venn.kdtas6in_vs_kdtas10_11_in.kdtas6po_vs_kdtas10_11_po.10",sep=''))
	dev.off()
}	

#c("contpo_vs_cont", "kdapo_vs_contpo", "kdepo_vs_contpo", "kdfpo_vs_contpo", "kdgpo_vs_contpo", "kdtas10_11_po_vs_kdtas10_11_in", "kdtas2po_vs_kdtas10_11_po", "kdtas4po_vs_kdtas10_11_po", "kdtas6po_vs_kdtas10_11_po", "kdamo_vs_cont", "kdemo_vs_cont", "kdfin_vs_cont", "kdgin_vs_cont", "kdtas10_11_in_vs_cont", "kdtas2in_vs_kdtas10_11_in", "kdtas4in_vs_kdtas10_11_in", "kdtas6in_vs_kdtas10_11_in", "kda_vs_cont", "kde_vs_cont") 

po_only_files<-c("2venn.kda_vs_cont.kdapo_vs_contpo.01.444.txt","2venn.kde_vs_cont.kdepo_vs_contpo.01.703.txt","2venn.kdtas2in_vs_kdtas10_11_in.kdtas2po_vs_kdtas10_11_po.01.89.txt", 
	"2venn.kdtas4in_vs_kdtas10_11_in.kdtas4po_vs_kdtas10_11_po.01.2549.txt", "2venn.kdtas6in_vs_kdtas10_11_in.kdtas6po_vs_kdtas10_11_po.01.346.txt","2venn.kdgin_vs_cont.kdgpo_vs_contpo.01.539.txt")

f<-list()
for(i in 1:length(po_only_files))
{
	f[[i]]<-read.table(po_only_files[i],sep='\t',as.is=T,quote="",comment.char="",header=T)[,1]
}

setnames<-c("kdapo_444","kdepo_703","kdtas2po_89","kdtas4_2549","kdtas6_346","kdgpo_539")
Vcombo<-Venn(SetNames=setnames[1:2],Sets=f[c(1,2)])

png("2venn.kdapo.kdepo.png",height=600,width=800,pointsize=16)
plot(Vcombo)
dev.off()

Vcombo<-Venn(SetNames=setnames[c(1,6)],Sets=f[c(1,6)])
#png("2venn.kdapo.kdepo.png",height=600,width=800,pointsize=16)
plot(Vcombo)
#dev.off()



updown_143<-rep("up",length(Vcombo@IntersectionSets$`10`))
updown_143[genes_de$kdapo_vs_contpo[match(Vcombo@IntersectionSets$`10`,genes_de$kdapo_vs_contpo[,1]),2]<0]<-"down"
writegenes_updown(Vcombo@IntersectionSets$`10`,updown_143,paste("2venn.6-26-2020.10.updown",sep=''))

updown_413<-rep("up",length(Vcombo@IntersectionSets$`01`))
updown_413[genes_de$kdepo_vs_contpo[match(Vcombo@IntersectionSets$`01`,genes_de$kdepo_vs_contpo[,1]),2]<0]<-"down"
writegenes_updown(Vcombo@IntersectionSets$`01`,updown_406,paste("2venn.6-26-2020.01.updown",sep=''))

updown_290_kdapo<-rep("up in kdapo",length(Vcombo@IntersectionSets$`11`))
updown_290_kdapo[genes_de$kdapo_vs_contpo[match(Vcombo@IntersectionSets$`11`,genes_de$kdapo_vs_contpo[,1]),2]<0]<-"down in kdapo"
updown_290_kdepo<-rep("up in kdepo",length(Vcombo@IntersectionSets$`11`))
updown_290_kdepo[genes_de$kdepo_vs_contpo[match(Vcombo@IntersectionSets$`11`,genes_de$kdepo_vs_contpo[,1]),2]<0]<-"down in kdepo"
updown_290<-paste(updown_290_kdapo,updown_290_kdepo,sep=" ")
updown_290[updown_290=="up in kdapo up in kdepo"]<-"up in kdapo and kdepo"
updown_290[updown_290=="down in kdapo down in kdepo"]<-"down in kdapo and kdepo"
updown_290[updown_290=="up in kdapo down in kdepo"]<-"up in kdapo, down in kdepo"
updown_290[updown_290=="down in kdapo up in kdepo"]<-"down in kdapo, up in kdepo"
writegenes_updown(Vcombo@IntersectionSets$`11`,updown_290,paste("2venn.6-26-2020.11.updown",sep=''))


#png("3venn.yellow_venns.png",height=600,width=800,pointsize=16)
#plot(Vcombo)
#dev.off()
#png("3venn.yellow_venns.chow_ruskey.png",height=600,width=800,pointsize=16)
#plot(Vcombo,type="ChowRuskey")
#dev.off()

writegenes_updown<-function(genes,updown,filenamepart)
{
	df<-genedata[genedata[,1] %in% genes,c("Gene_ID","Name","Biotype","Description")]
	df$updown<-updown
	both.iv<-grepl("up.*down.*",updown) | grepl("down.*up.*",updown)
	if(length(which(both.iv))>0)
	{
		write.table(df[both.iv,],paste(filenamepart,".",length(df[,1]),".",length(df[both.iv,1]),".both.txt",sep=''),row.names=F,quote=F,sep='\t')
	}
	up.iv<-grepl("up",updown) & !both.iv
	dn.iv<-grepl("down",updown) & !both.iv
	write.table(df[up.iv,],paste(filenamepart,".",length(df[,1]),".",length(df[up.iv,1]),".up.txt",sep=''),row.names=F,quote=F,sep='\t')
	write.table(df[dn.iv,],paste(filenamepart,".",length(df[,1]),".",length(df[dn.iv,1]),".dn.txt",sep=''),row.names=F,quote=F,sep='\t')
}

updown_46<-rep("up",length(Vcombo@IntersectionSets$`100`))
updown_46[genes_de$kdtas2po_vs_kdtas10_11_po[match(Vcombo@IntersectionSets$`100`,genes_de$kdtas2po_vs_kdtas10_11_po[,1]),2]<0]<-"down"
writegenes_updown(Vcombo@IntersectionSets$`100`,updown_46,paste("3venn.yellowvenn.100.updown",sep=''))

updown_247<-rep("up",length(Vcombo@IntersectionSets$`001`))
updown_247[genes_de$kdtas6po_vs_kdtas10_11_po[match(Vcombo@IntersectionSets$`001`,genes_de$kdtas6po_vs_kdtas10_11_po[,1]),2]<0]<-"down"
writegenes_updown(Vcombo@IntersectionSets$`001`,updown_247,paste("3venn.yellowvenn.001.updown",sep=''))

updown_10_tas2<-rep("up in kdtas2",length(Vcombo@IntersectionSets$`111`))
updown_10_tas2[genes_de$kdtas2po_vs_kdtas10_11_po[match(Vcombo@IntersectionSets$`111`,genes_de$kdtas2po_vs_kdtas10_11_po[,1]),2]<0]<-"down in kdtas2"
updown_10_tas6<-rep("up in kdtas6",length(Vcombo@IntersectionSets$`111`))
updown_10_tas6[genes_de$kdtas6po_vs_kdtas10_11_po[match(Vcombo@IntersectionSets$`111`,genes_de$kdtas6po_vs_kdtas10_11_po[,1]),2]<0]<-"down in kdtas6"
updown_10_tas4<-rep("up in kdtas4",length(Vcombo@IntersectionSets$`111`))
updown_10_tas4[genes_de$kdtas4po_vs_kdtas10_11_po[match(Vcombo@IntersectionSets$`111`,genes_de$kdtas4po_vs_kdtas10_11_po[,1]),2]<0]<-"down in kdtas4"
updown_10<-rep("up in all",length(Vcombo@IntersectionSets$`111`))
updown_10[genes_de$kdtas2po_vs_kdtas10_11_po[match(Vcombo@IntersectionSets$`111`,genes_de$kdtas2po_vs_kdtas10_11_po[,1]),2]<0]<-"down in all"
writegenes_updown(Vcombo@IntersectionSets$`111`,updown_10,paste("3venn.yellowvenn.111.updown",sep=''))

updown_2447<-rep("up",length(Vcombo@IntersectionSets$`010`))
updown_2447[genes_de$kdtas4po_vs_kdtas10_11_po[match(Vcombo@IntersectionSets$`010`,genes_de$kdtas4po_vs_kdtas10_11_po[,1]),2]<0]<-"down"
writegenes_updown(Vcombo@IntersectionSets$`010`,updown_2447,paste("3venn.yellowvenn.010.updown",sep=''))

updown_18_tas2<-rep("up in kdtas2",length(Vcombo@IntersectionSets$`110`))
updown_18_tas2[genes_de$kdtas2po_vs_kdtas10_11_po[match(Vcombo@IntersectionSets$`110`,genes_de$kdtas2po_vs_kdtas10_11_po[,1]),2]<0]<-"down in kdtas2"
updown_18_tas4<-rep("up in kdtas4",length(Vcombo@IntersectionSets$`110`))
updown_18_tas4[genes_de$kdtas4po_vs_kdtas10_11_po[match(Vcombo@IntersectionSets$`110`,genes_de$kdtas4po_vs_kdtas10_11_po[,1]),2]<0]<-"down in kdtas4"
updown_18<-rep("up in kdtas2 and kdtas4",length(Vcombo@IntersectionSets$`110`))
updown_18[genes_de$kdtas2po_vs_kdtas10_11_po[match(Vcombo@IntersectionSets$`110`,genes_de$kdtas2po_vs_kdtas10_11_po[,1]),2]<0]<-"down in kdtas2 and kdtas4"
writegenes_updown(Vcombo@IntersectionSets$`110`,updown_18,paste("3venn.yellowvenn.110.updown",sep=''))

updown_15_tas2<-rep("up in kdtas2",length(Vcombo@IntersectionSets$`101`))
updown_15_tas2[genes_de$kdtas2po_vs_kdtas10_11_po[match(Vcombo@IntersectionSets$`101`,genes_de$kdtas2po_vs_kdtas10_11_po[,1]),2]<0]<-"down in kdtas2"
updown_15_tas6<-rep("up in kdtas6",length(Vcombo@IntersectionSets$`101`))
updown_15_tas6[genes_de$kdtas6po_vs_kdtas10_11_po[match(Vcombo@IntersectionSets$`101`,genes_de$kdtas6po_vs_kdtas10_11_po[,1]),2]<0]<-"down in kdtas6"
updown_15<-rep("up in kdtas2 and kdtas6",length(Vcombo@IntersectionSets$`101`))
updown_15[genes_de$kdtas2po_vs_kdtas10_11_po[match(Vcombo@IntersectionSets$`101`,genes_de$kdtas2po_vs_kdtas10_11_po[,1]),2]<0]<-"down in kdtas2 and kdtas6"
writegenes_updown(Vcombo@IntersectionSets$`101`,updown_15,paste("3venn.yellowvenn.101.updown",sep=''))

updown_74_tas4<-rep("up in kdtas4",length(Vcombo@IntersectionSets$`011`))
updown_74_tas4[genes_de$kdtas4po_vs_kdtas10_11_po[match(Vcombo@IntersectionSets$`011`,genes_de$kdtas4po_vs_kdtas10_11_po[,1]),2]<0]<-"down in kdtas4"
updown_74_tas6<-rep("up in kdtas6",length(Vcombo@IntersectionSets$`011`))
updown_74_tas6[genes_de$kdtas6po_vs_kdtas10_11_po[match(Vcombo@IntersectionSets$`011`,genes_de$kdtas6po_vs_kdtas10_11_po[,1]),2]<0]<-"down in kdtas6"
updown_74<-paste(updown_74_tas4,updown_74_tas6,sep=" ")
updown_74[updown_74=="up in kdtas4 up in kdtas6"]<-"up in kdtas4 and kdtas6"
updown_74[updown_74=="down in kdtas4 down in kdtas6"]<-"down in kdtas4 and kdtas6"
updown_74[updown_74=="up in kdtas4 down in kdtas6"]<-"up in kdtas4, down in kdtas6"
updown_74[updown_74=="down in kdtas4 up in kdtas6"]<-"down in kdtas4, up in kdtas6"
writegenes_updown(Vcombo@IntersectionSets$`011`,updown_74,paste("3venn.yellowvenn.011.updown",sep=''))

po_only_files<-c("2venn.kdtas4in_vs_kdtas10_11_in.kdtas4po_vs_kdtas10_11_po.01.2549.txt", "2venn.kdtas6in_vs_kdtas10_11_in.kdtas6po_vs_kdtas10_11_po.01.346.txt")
f<-list()
for(i in 1:length(po_only_files))
{
	f[[i]]<-read.table(po_only_files[i],sep='\t',as.is=T,quote="",comment.char="",header=T)[,1]
}
setnames<-c("kdtas4_2549","kdtas6_346")
Vcombo<-Venn(SetNames=setnames,Sets=f)
writegenes(Vcombo@IntersectionSets$`10`,paste("2venn.yellowvenn.10",sep=''))
writegenes(Vcombo@IntersectionSets$`11`,paste("2venn.yellowvenn.11",sep=''))
writegenes(Vcombo@IntersectionSets$`01`,paste("2venn.yellowvenn.01",sep=''))

png("2venn.yellow_venns.png",height=600,width=800,pointsize=16)
plot(Vcombo)
dev.off()

#kdapo and kdepo, added  6/26/2020
po_only_files<-c("2venn.kdapo_vs_contpo_11_in.kdtas4po_vs_kdtas10_11_po.01.2549.txt", "2venn.kdtas6in_vs_kdtas10_11_in.kdtas6po_vs_kdtas10_11_po.01.346.txt")
f<-list()
for(i in 1:length(po_only_files))
{
	f[[i]]<-read.table(po_only_files[i],sep='\t',as.is=T,quote="",comment.char="",header=T)[,1]
}
setnames<-c("kdtas4_2549","kdtas6_346")
Vcombo<-Venn(SetNames=setnames,Sets=f)
writegenes(Vcombo@IntersectionSets$`10`,paste("2venn.yellowvenn.10",sep=''))
writegenes(Vcombo@IntersectionSets$`11`,paste("2venn.yellowvenn.11",sep=''))
writegenes(Vcombo@IntersectionSets$`01`,paste("2venn.yellowvenn.01",sep=''))

png("2venn.yellow_venns.png",height=600,width=800,pointsize=16)
plot(Vcombo)
dev.off()
```

Venn diagrams for tas samples, input vs. polysome.

```{r,out.width="600px"}
knitr::include_graphics("../data/2venn.kdtas2in_vs_kdtas10_11_in.kdtas2po_vs_kdtas10_11_po.png")
knitr::include_graphics("../data/2venn.kdtas4in_vs_kdtas10_11_in.kdtas4po_vs_kdtas10_11_po.png")
knitr::include_graphics("../data/2venn.kdtas6in_vs_kdtas10_11_in.kdtas6po_vs_kdtas10_11_po.png")
```

Chow-ruskey style venn of all the yellow parts of the venn diagrams (polysome-only de genes) for kdtas2, kdtas4, kdtas6.

```{r,out.width="600px"}
knitr::include_graphics("../data/3venn.yellow_venns.chow_ruskey.png")
```

Area proportional venn

```{r,out.width="600px"}
knitr::include_graphics("../data/3venn.yellow_venns.png")
```

There are 10 genes that are present in all the selections:  
[3venn.yellowvenn.111.10.txt](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.111.10.txt)  

15 genes in tas6po and tas2po:  
[3venn.yellowvenn.101.15.txt](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.101.15.txt)  

18 genes in tas4po and tas2po:  
[3venn.yellowvenn.110.18.txt](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.110.18.txt)  

74 genes in tas4po and tas6po:  
[3venn.yellowvenn.011.74.txt](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.011.74.txt)  

46 genes in tas2po only:  
[3venn.yellowvenn.100.46.txt](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.100.46.txt)  

247 genes in tas6po only:  
[3venn.yellowvenn.001.247.txt](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.001.247.txt)  

2447 genes in tas4po only:  
[3venn.yellowvenn.010.2447.txt](https://webfs/n/core/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.010.2447.txt)  

### Metagene Plots

```{r metagene, echo=T, message=F,results="hide",warning=F,eval=F}
library(Rsamtools)
library(GenomicAlignments)
library(GenomicRanges)
library(rtracklayer)


bed<-read.table("/n/data1/genomes/indexes/hg38/annotation/Ens_91/beds/hg38.Ens_91.genes.bed",sep='\t',skip=1,as.is=T)
colnames(bed)<-c("chrom.gene","start.gene","end.gene","name.gene","score.gene","strand.gene")
goi<-bed

tss<-read.table("hg38.Ens_91.tss.bed",sep='\t',as.is=T,skip=1)
colnames(tss)<-c("chrom","start","end","name","score","strand")
tss_regions<-list()
tss_regions_norm<-list()

tes<-read.table("hg38.Ens_91.tes.bed",sep='\t',as.is=T,skip=1)
colnames(tes)<-c("chrom","start","end","name","score","strand")
tes_regions<-list()
tes_regions_norm<-list()

chromlen<-read.table("/n/data1/genomes/indexes/hg38/hg38.fa.fai",as.is=T)
colnames(chromlen)<-c("chrom","length")
endcheck <- function(chrom,val)
{
   #if start is less than 0 return 0
   if(val < 1)
   {     
      ret <- 1;
   }  
   #if end is past end of chrom return end of chrom
   else if(chromlen[chromlen[,1]==chrom,"length"] < val)
   {
      ret <- chromlen[chromlen[,1]==chrom,"length"];
   }     
   else  
   {  
      ret<-val
   }
   return(ret);
}

#Binning expression over body of gene +/- 1kb to look for positional effects (3' end?)
avgs<-list()
avgs_norm<-list()
for(i in 1:length(targets[,1]))
{
	avgs[[i]]<-data.frame(matrix(nrow=length(goi[,1]),ncol=81))
	avgs[[i]][,1]<-goi[,4]

	avgs_norm[[i]]<-data.frame(matrix(nrow=length(goi[,1]),ncol=81))
	avgs_norm[[i]][,1]<-goi[,4]

	tss_regions[[i]]<-data.frame(matrix(nrow=length(goi[,1]),ncol=121))
	tss_regions[[i]][,1]<-goi[,4]
	tss_regions_norm[[i]]<-data.frame(matrix(nrow=length(goi[,1]),ncol=121))
	tss_regions_norm[[i]][,1]<-goi[,4]

	tes_regions[[i]]<-data.frame(matrix(nrow=length(goi[,1]),ncol=121))
	tes_regions[[i]][,1]<-goi[,4]
	tes_regions_norm[[i]]<-data.frame(matrix(nrow=length(goi[,1]),ncol=121))
	tes_regions_norm[[i]][,1]<-goi[,4]

	cur<-readGAlignments(paste("/n/core/Bioinformatics/secondary/Workman/tas/",targets$Order[i],"/",targets$alias[i],"/accepted_hits.bam",sep=''))

	#flip the strands
	cvg_pos<-coverage(cur[strand(cur)=="-"])
	cvg_neg<-coverage(cur[strand(cur)=="+"])
	newcvg_pos<-(cvg_pos*1e6)/bam_stats[i,"total_alignments"]
	newcvg_neg<-(cvg_neg*1e6)/bam_stats[i,"total_alignments"]

	for(j in 1:length(goi[,1]))
	{
		if(goi$strand.gene[j] == "+")
		{
			upstream <- window(newcvg_pos[[goi$chrom[j]]],endcheck(goi$chrom[j],goi$start.gene[j]-1000),endcheck(goi$chrom[j],goi$start.gene[j]))
			genebody <- window(newcvg_pos[[goi$chrom[j]]],endcheck(goi$chrom[j],goi$start.gene[j]),endcheck(goi$chrom[j],goi$end.gene[j]))
			downstream <- window(newcvg_pos[[goi$chrom[j]]],endcheck(goi$chrom[j],goi$end.gene[j]),endcheck(goi$chrom[j],goi$end.gene[j]+1000))

			tss_regions[[i]][j,2:121]<-approx(window(newcvg_pos[[goi$chrom[j]]],endcheck(goi$chrom[j],goi$start[j]-1000),endcheck(goi$chrom[j],goi$start[j]+1000)),n=120)$y
			tss_regions_norm[[i]][j,2:121]<-tss_regions[[i]][j,2:121]/sum(tss_regions[[i]][j,2:121])

			tes_regions[[i]][j,2:121]<-approx(window(newcvg_pos[[goi$chrom[j]]],endcheck(goi$chrom[j],goi$end[j]-1000),endcheck(goi$chrom[j],goi$end[j]+1000)),n=120)$y
			tes_regions_norm[[i]][j,2:121]<-tes_regions[[i]][j,2:121]/sum(tes_regions[[i]][j,2:121])

         if(length(upstream)>1 & length(genebody) > 1 & length(downstream) > 1)
         {
            avgs[[i]][j,2:21]<-approx(upstream,n=20)$y
            avgs[[i]][j,22:61]<-approx(genebody,n=40)$y
            avgs[[i]][j,62:81]<-approx(downstream,n=20)$y

				avgs_norm[[i]][j,2:81]<-avgs[[i]][j,2:81]/sum(avgs[[i]][j,2:81])
         }
         else
         {
            avgs[[i]][j,2:81]<-NA
				avgs_norm[[i]][j,2:81]<-NA
         }
		} else if (goi$strand.gene[j]=="-")
		{
			upstream <- window(newcvg_neg[[goi$chrom[j]]],endcheck(goi$chrom[j],goi$start.gene[j]-1000),endcheck(goi$chrom[j],goi$start.gene[j]))
			genebody <- window(newcvg_neg[[goi$chrom[j]]],endcheck(goi$chrom[j],goi$start.gene[j]),endcheck(goi$chrom[j],goi$end.gene[j]))
			downstream <- window(newcvg_neg[[goi$chrom[j]]],endcheck(goi$chrom[j],goi$end.gene[j]),endcheck(goi$chrom[j],goi$end.gene[j]+1000))

			tes_regions[[i]][j,2:121]<-approx(rev(window(newcvg_neg[[goi$chrom[j]]],endcheck(goi$chrom[j],goi$start[j]-1000),endcheck(goi$chrom[j],goi$start[j]+1000))),n=120)$y
			tes_regions_norm[[i]][j,2:121]<-tes_regions[[i]][j,2:121]/sum(tes_regions[[i]][j,2:121])

			tss_regions[[i]][j,2:121]<-approx(rev(window(newcvg_neg[[goi$chrom[j]]],endcheck(goi$chrom[j],goi$end[j]-1000),endcheck(goi$chrom[j],goi$end[j]+1000))),n=120)$y
			tss_regions_norm[[i]][j,2:121]<-tss_regions[[i]][j,2:121]/sum(tss_regions[[i]][j,2:121])

         if(length(upstream)>1 & length(genebody) > 1 & length(downstream) > 1)
         {
            avgs[[i]][j,2:21]<-approx(rev(downstream),n=20)$y
            avgs[[i]][j,22:61]<-approx(rev(genebody),n=40)$y
            avgs[[i]][j,62:81]<-approx(rev(upstream),n=20)$y

				avgs_norm[[i]][j,2:81]<-avgs[[i]][j,2:81]/sum(avgs[[i]][j,2:81])
         }
         else
         {
            avgs[[i]][j,2:81]<-NA
				avgs_norm[[i]][j,2:81]<-NA
         }
		}
	}
}

save(avgs,file="avgs.Rdata")
save(avgs_norm,file="avgs_norm.Rdata")
save(tss_regions,file="tss_regions.Rdata")
save(tss_regions_norm,file="tss_regions_norm.Rdata")
save(tes_regions,file="tes_regions.Rdata")
save(tes_regions_norm,file="tes_regions_norm.Rdata")
```

``` {r metagene plots, echo=TRUE, message=FALSE, results="hide", warning=FALSE, eval=FALSE}
load("avgs.Rdata")
load("avgs_norm.Rdata")
load("tss_regions.Rdata")
load("tss_regions_norm.Rdata")
load("tes_regions.Rdata")
load("tes_regions_norm.Rdata")

targets$group<-c("s_cont", "s_cont", "s_cont", "s_kda", "s_kda", "s_kda", "s_kdb", "s_kdb", "s_kdb", "s_kdc", "s_kdc", "s_kdc", "s_kde", "s_kde", "s_kde", "s_contmo", "s_contmo", "s_contpo", "s_contpo", "s_kdamo", "s_kdamo", "s_kdemo", "s_kdemo", "s_kdfmo", "s_kdfmo", "s_kdgmo", "s_kdgmo", "s_kdapo", "s_kdapo", "s_kdepo", "s_kdepo", "s_kdfpo", "s_kdfpo", "s_kdgpo", "s_kdgpo", "s_kdfin", "s_kdfin", "s_kdgin", "s_kdgin", "s_kdtas2mo", "s_kdtas2mo", "s_kdtas4mo", "s_kdtas4mo", "s_kdtas6mo", "s_kdtas6mo", "s_kdtas10mo", "s_kdtas10mo", "s_kdtas11mo", "s_kdtas11mo", "s_kdtas2po", "s_kdtas2po", "s_kdtas4po", "s_kdtas4po", "s_kdtas6po", "s_kdtas6po", "s_kdtas10po", "s_kdtas10po", "s_kdtas11po", "s_kdtas11po", "s_kdtas2in", "s_kdtas2in", "s_kdtas4in", "s_kdtas4in", "s_kdtas6in", "s_kdtas6in", "s_kdtas10in", "s_kdtas10in", "s_kdtas11in", "s_kdtas11in")

#pdf("metagene_plots.pdf",height=8,width=20)
png("metagene_plots.png",height=800,width=2000,pointsize=16)
par(mfrow=c(4,8))
for(i in seq(from=1,to=13,by=3))
{
	ylimmin<-min(c(colMeans(avgs[[i]][,2:81]),colMeans(avgs[[i+1]][,2:81]),colMeans(avgs[[i+2]][,2:81])),na.rm=T)
	ylimmax<-1.4*max(c(colMeans(avgs[[i]][,2:81]),colMeans(avgs[[i+1]][,2:81]),colMeans(avgs[[i+2]][,2:81])),na.rm=T)
	plot(colMeans(avgs[[i]][,2:81],na.rm=T),ylim=c(ylimmin,ylimmax),type='l',col=cols[i],main=targets$group[i],ylab="average expression value")
	lines(colMeans(avgs[[i+1]][,2:81],na.rm=T),lty=2,col=cols[i])
	lines(colMeans(avgs[[i+2]][,2:81],na.rm=T),lty=3,col=cols[i])
	legend("topright",legend=targets$Sample.Name[i:(i+2)],lty=1:3,col=cols[i],cex=.6)
}
for(i in seq(from=16,to=68,by=2))
{
	ylimmin<-min(c(colMeans(avgs[[i]][,2:81]),colMeans(avgs[[i+1]][,2:81])),na.rm=T)
	ylimmax<-1.4*max(c(colMeans(avgs[[i]][,2:81]),colMeans(avgs[[i+1]][,2:81])),na.rm=T)
	plot(colMeans(avgs[[i]][,2:81],na.rm=T),ylim=c(ylimmin,ylimmax),type='l',col=cols[i],main=targets$group[i],ylab="average expression value")
	lines(colMeans(avgs[[i+1]][,2:81],na.rm=T),lty=2,col=cols[i])
	legend("topright",legend=targets$Sample.Name[i:(i+1)],lty=1:2,col=cols[i],cex=.6)
}
dev.off()

pdf("metagene_plots.norm.pdf",height=8,width=20)
#png("metagene_plots.norm.png",height=800,width=2000,pointsize=16)
par(mfrow=c(4,8))
for(i in seq(from=1,to=13,by=3))
{
	ylimmin<-0
	ylimmax<-.08
	#ylimmin<-min(c(colMeans(avgs_norm[[i]][,2:81],na.rm=T),colMeans(avgs_norm[[i+1]][,2:81],na.rm=T),colMeans(avgs_norm[[i+2]][,2:81],na.rm=T)),na.rm=T)
	#ylimmax<-1.4*max(c(colMeans(avgs_norm[[i]][,2:81],na.rm=T),colMeans(avgs_norm[[i+1]][,2:81],na.rm=T),colMeans(avgs_norm[[i+2]][,2:81],na.rm=T)),na.rm=T)
	plot(colMeans(avgs_norm[[i]][,2:81],na.rm=T),ylim=c(ylimmin,ylimmax),type='l',col=cols[i],main=targets$group[i],ylab="average expression value")
	lines(colMeans(avgs_norm[[i+1]][,2:81],na.rm=T),lty=2,col=cols[i])
	lines(colMeans(avgs_norm[[i+2]][,2:81],na.rm=T),lty=3,col=cols[i])
	legend("topright",legend=targets$Sample.Name[i:(i+2)],lty=1:3,col=cols[i],cex=.6)
}
for(i in seq(from=16,to=68,by=2))
{
	ylimmin<-0
	ylimmax<-.08
	#ylimmin<-min(c(colMeans(avgs_norm[[i]][,2:81],na.rm=T),colMeans(avgs_norm[[i+1]][,2:81],na.rm=T)),na.rm=T)
	#ylimmax<-1.4*max(c(colMeans(avgs_norm[[i]][,2:81],na.rm=T),colMeans(avgs_norm[[i+1]][,2:81],na.rm=T)),na.rm=T)
	plot(colMeans(avgs_norm[[i]][,2:81],na.rm=T),ylim=c(ylimmin,ylimmax),type='l',col=cols[i],main=targets$group[i],ylab="average expression value")
	lines(colMeans(avgs_norm[[i+1]][,2:81],na.rm=T),lty=2,col=cols[i])
	legend("topright",legend=targets$Sample.Name[i:(i+1)],lty=1:2,col=cols[i],cex=.6)
}
dev.off()

pdf("tss_plots.pdf",height=8,width=20)
#png("tss_plots.png",height=800,width=2000,pointsize=16)
par(mfrow=c(4,8))
for(i in seq(from=1,to=13,by=3))
{
	ylimmin<-min(c(colMeans(tss_regions[[i]][,2:121]),colMeans(tss_regions[[i+1]][,2:121]),colMeans(tss_regions[[i+2]][,2:121])),na.rm=T)
	ylimmax<-1.4*max(c(colMeans(tss_regions[[i]][,2:121]),colMeans(tss_regions[[i+1]][,2:121]),colMeans(tss_regions[[i+2]][,2:121])),na.rm=T)
	plot(colMeans(tss_regions[[i]][,2:121],na.rm=T),ylim=c(ylimmin,ylimmax),type='l',col=cols[i],main=targets$group[i],ylab="average expression value")
	lines(colMeans(tss_regions[[i+1]][,2:121],na.rm=T),lty=2,col=cols[i])
	lines(colMeans(tss_regions[[i+2]][,2:121],na.rm=T),lty=3,col=cols[i])
	legend("topright",legend=targets$Sample.Name[i:(i+2)],lty=1:3,col=cols[i],cex=.6)
}
for(i in seq(from=16,to=68,by=2))
{
	ylimmin<-min(c(colMeans(tss_regions[[i]][,2:121]),colMeans(tss_regions[[i+1]][,2:121])),na.rm=T)
	ylimmax<-1.4*max(c(colMeans(tss_regions[[i]][,2:121]),colMeans(tss_regions[[i+1]][,2:121])),na.rm=T)
	plot(colMeans(tss_regions[[i]][,2:121],na.rm=T),ylim=c(ylimmin,ylimmax),type='l',col=cols[i],main=targets$group[i],ylab="average expression value")
	lines(colMeans(tss_regions[[i+1]][,2:121],na.rm=T),lty=2,col=cols[i])
	legend("topright",legend=targets$Sample.Name[i:(i+1)],lty=1:2,col=cols[i],cex=.6)
}
dev.off()

pdf("tss_plots.norm.pdf",height=8,width=20)
#png("tss_plots.norm.png",height=800,width=2000,pointsize=16)
par(mfrow=c(4,8))
for(i in seq(from=1,to=13,by=3))
{
	ylimmin<-0
	ylimmax<-.03
	#ylimmin<-min(c(colMeans(tss_regions_norm[[i]][,2:121],na.rm=T),colMeans(tss_regions_norm[[i+1]][,2:121],na.rm=T),colMeans(tss_regions_norm[[i+2]][,2:121],na.rm=T)),na.rm=T)
	#ylimmax<-1.4*max(c(colMeans(tss_regions_norm[[i]][,2:121],na.rm=T),colMeans(tss_regions_norm[[i+1]][,2:121],na.rm=T),colMeans(tss_regions_norm[[i+2]][,2:121],na.rm=T)),na.rm=T)
	plot(colMeans(tss_regions_norm[[i]][,2:121],na.rm=T),ylim=c(ylimmin,ylimmax),type='l',col=cols[i],main=targets$group[i],ylab="average expression value")
	lines(colMeans(tss_regions_norm[[i+1]][,2:121],na.rm=T),lty=2,col=cols[i])
	lines(colMeans(tss_regions_norm[[i+2]][,2:121],na.rm=T),lty=3,col=cols[i])
	legend("topright",legend=targets$Sample.Name[i:(i+2)],lty=1:3,col=cols[i],cex=.6)
}
for(i in seq(from=16,to=68,by=2))
{
	ylimmin<-0
	ylimmax<-.03
	#ylimmin<-min(c(colMeans(tss_regions_norm[[i]][,2:121],na.rm=T),colMeans(tss_regions_norm[[i+1]][,2:121],na.rm=T)),na.rm=T)
	#ylimmax<-1.4*max(c(colMeans(tss_regions_norm[[i]][,2:121],na.rm=T),colMeans(tss_regions_norm[[i+1]][,2:121],na.rm=T)),na.rm=T)
	plot(colMeans(tss_regions_norm[[i]][,2:121],na.rm=T),ylim=c(ylimmin,ylimmax),type='l',col=cols[i],main=targets$group[i],ylab="average expression value")
	lines(colMeans(tss_regions_norm[[i+1]][,2:121],na.rm=T),lty=2,col=cols[i])
	legend("topright",legend=targets$Sample.Name[i:(i+1)],lty=1:2,col=cols[i],cex=.6)
}
dev.off()

pdf("tes_plots.pdf",height=8,width=20)
#png("tes_plots.png",height=800,width=2000,pointsize=16)
par(mfrow=c(4,8))
for(i in seq(from=1,to=13,by=3))
{
	ylimmin<-min(c(colMeans(tes_regions[[i]][,2:121]),colMeans(tes_regions[[i+1]][,2:121]),colMeans(tes_regions[[i+2]][,2:121])),na.rm=T)
	ylimmax<-1.4*max(c(colMeans(tes_regions[[i]][,2:121]),colMeans(tes_regions[[i+1]][,2:121]),colMeans(tes_regions[[i+2]][,2:121])),na.rm=T)
	plot(colMeans(tes_regions[[i]][,2:121],na.rm=T),ylim=c(ylimmin,ylimmax),type='l',col=cols[i],main=targets$group[i],ylab="average expression value")
	lines(colMeans(tes_regions[[i+1]][,2:121],na.rm=T),lty=2,col=cols[i])
	lines(colMeans(tes_regions[[i+2]][,2:121],na.rm=T),lty=3,col=cols[i])
	legend("topright",legend=targets$Sample.Name[i:(i+2)],lty=1:3,col=cols[i],cex=.6)
}
for(i in seq(from=16,to=68,by=2))
{
	ylimmin<-min(c(colMeans(tes_regions[[i]][,2:121]),colMeans(tes_regions[[i+1]][,2:121])),na.rm=T)
	ylimmax<-1.4*max(c(colMeans(tes_regions[[i]][,2:121]),colMeans(tes_regions[[i+1]][,2:121])),na.rm=T)
	plot(colMeans(tes_regions[[i]][,2:121],na.rm=T),ylim=c(ylimmin,ylimmax),type='l',col=cols[i],main=targets$group[i],ylab="average expression value")
	lines(colMeans(tes_regions[[i+1]][,2:121],na.rm=T),lty=2,col=cols[i])
	legend("topright",legend=targets$Sample.Name[i:(i+1)],lty=1:2,col=cols[i],cex=.6)
}
dev.off()

pdf("tes_plots.norm.pdf",height=8,width=20)
#png("tes_plots.norm.png",height=800,width=2000,pointsize=16)
par(mfrow=c(4,8))
for(i in seq(from=1,to=13,by=3))
{
	ylimmin<-0
	ylimmax<-.03
	#ylimmin<-min(c(colMeans(tes_regions_norm[[i]][,2:121],na.rm=T),colMeans(tes_regions_norm[[i+1]][,2:121],na.rm=T),colMeans(tes_regions_norm[[i+2]][,2:121],na.rm=T)),na.rm=T)
	#ylimmax<-1.4*max(c(colMeans(tes_regions_norm[[i]][,2:121],na.rm=T),colMeans(tes_regions_norm[[i+1]][,2:121],na.rm=T),colMeans(tes_regions_norm[[i+2]][,2:121],na.rm=T)),na.rm=T)
	plot(colMeans(tes_regions_norm[[i]][,2:121],na.rm=T),ylim=c(ylimmin,ylimmax),type='l',col=cols[i],main=targets$group[i],ylab="average expression value")
	lines(colMeans(tes_regions_norm[[i+1]][,2:121],na.rm=T),lty=2,col=cols[i])
	lines(colMeans(tes_regions_norm[[i+2]][,2:121],na.rm=T),lty=3,col=cols[i])
	legend("topright",legend=targets$Sample.Name[i:(i+2)],lty=1:3,col=cols[i],cex=.6)
}
for(i in seq(from=16,to=68,by=2))
{
	ylimmin<-0
	ylimmax<-.03
	#ylimmin<-min(c(colMeans(tes_regions_norm[[i]][,2:121],na.rm=T),colMeans(tes_regions_norm[[i+1]][,2:121],na.rm=T)),na.rm=T)
	#ylimmax<-1.4*max(c(colMeans(tes_regions_norm[[i]][,2:121],na.rm=T),colMeans(tes_regions_norm[[i+1]][,2:121],na.rm=T)),na.rm=T)
	plot(colMeans(tes_regions_norm[[i]][,2:121],na.rm=T),ylim=c(ylimmin,ylimmax),type='l',col=cols[i],main=targets$group[i],ylab="average expression value")
	lines(colMeans(tes_regions_norm[[i+1]][,2:121],na.rm=T),lty=2,col=cols[i])
	legend("topright",legend=targets$Sample.Name[i:(i+1)],lty=1:2,col=cols[i],cex=.6)
}
dev.off()



#metagene plots of all genes, but one with kdg, kda, kde, kdg, cont. One with kdtas2,4,6,and average of 10 and 11. Full gene, TSS, TES.
names(avgs_norm)<-names(tss_regions_norm)<-names(tes_regions_norm)<-targets$alias
names(cols)<-targets$alias
kda<-colMeans(rbind(colMeans(avgs_norm[["s_kda1"]][,2:81],na.rm=T),colMeans(avgs_norm[["s_kda2"]][,2:81],na.rm=T),colMeans(avgs_norm[["s_kda4"]][,2:81],na.rm=T)))
kde<-colMeans(rbind(colMeans(avgs_norm[["s_kde1"]][,2:81],na.rm=T),colMeans(avgs_norm[["s_kde2"]][,2:81],na.rm=T),colMeans(avgs_norm[["s_kde3"]][,2:81],na.rm=T)))
kdgpo<-colMeans(rbind(colMeans(avgs_norm[["s_kdgpo1"]][,2:81],na.rm=T),colMeans(avgs_norm[["s_kdgpo2"]][,2:81],na.rm=T)))
cont<-colMeans(rbind(colMeans(avgs_norm[["s_cont1"]][,2:81],na.rm=T),colMeans(avgs_norm[["s_cont2"]][,2:81],na.rm=T),colMeans(avgs_norm[["s_cont3"]][,2:81],na.rm=T)))

kda_tss<-colMeans(rbind(colMeans(tss_regions_norm[["s_kda1"]][,2:121],na.rm=T),colMeans(tss_regions_norm[["s_kda2"]][,2:121],na.rm=T),colMeans(tss_regions_norm[["s_kda4"]][,2:121],na.rm=T)))
kde_tss<-colMeans(rbind(colMeans(tss_regions_norm[["s_kde1"]][,2:121],na.rm=T),colMeans(tss_regions_norm[["s_kde2"]][,2:121],na.rm=T),colMeans(tss_regions_norm[["s_kde3"]][,2:121],na.rm=T)))
kdgpo_tss<-colMeans(rbind(colMeans(tss_regions_norm[["s_kdgpo1"]][,2:121],na.rm=T),colMeans(tss_regions_norm[["s_kdgpo2"]][,2:121],na.rm=T)))
cont_tss<-colMeans(rbind(colMeans(tss_regions_norm[["s_cont1"]][,2:121],na.rm=T),colMeans(tss_regions_norm[["s_cont2"]][,2:121],na.rm=T),colMeans(tss_regions_norm[["s_cont3"]][,2:121],na.rm=T)))

kda_tes<-colMeans(rbind(colMeans(tes_regions_norm[["s_kda1"]][,2:121],na.rm=T),colMeans(tes_regions_norm[["s_kda2"]][,2:121],na.rm=T),colMeans(tes_regions_norm[["s_kda4"]][,2:121],na.rm=T)))
kde_tes<-colMeans(rbind(colMeans(tes_regions_norm[["s_kde1"]][,2:121],na.rm=T),colMeans(tes_regions_norm[["s_kde2"]][,2:121],na.rm=T),colMeans(tes_regions_norm[["s_kde3"]][,2:121],na.rm=T)))
kdgpo_tes<-colMeans(rbind(colMeans(tes_regions_norm[["s_kdgpo1"]][,2:121],na.rm=T),colMeans(tes_regions_norm[["s_kdgpo2"]][,2:121],na.rm=T)))
cont_tes<-colMeans(rbind(colMeans(tes_regions_norm[["s_cont1"]][,2:121],na.rm=T),colMeans(tes_regions_norm[["s_cont2"]][,2:121],na.rm=T),colMeans(tes_regions_norm[["s_cont3"]][,2:121],na.rm=T)))

png("lineplots.kda-g.norm.png",height=800,width=2400,pointsize=16)
par(mfrow=c(1,3))
plot(kda,type='l',col=cols["s_kda1"],ylim=c(.005,.06),lty=1,main="Gene +/- 1kb",lwd=2,cex.main=2,cex.axis=1.5,ylab="normalized average expression values")
lines(kde,col=cols["s_kde1"],lty=2,lwd=2)
lines(kdgpo,col=cols["s_kdgpo1"],lty=3,lwd=2)
lines(cont,col=cols["s_cont1"],lty=4,lwd=2)
legend("topleft",legend=c("kda","kde","kdgpo","cont"),col=cols[c("s_kda1","s_kde1","s_kdgpo1","s_cont1")],lty=1:4,lwd=2,cex=2)

plot(kda_tss,type='l',col=cols["s_kda1"],ylim=c(0,.03),lty=1,main="TSS +/- 1kb",lwd=2,cex.main=2,cex.axis=1.5,ylab="")
lines(kde_tss,col=cols["s_kde1"],lty=2,lwd=2)
lines(kdgpo_tss,col=cols["s_kdgpo1"],lty=3,lwd=2)
lines(cont_tss,col=cols["s_cont1"],lty=4,lwd=2)
legend("topleft",legend=c("kdaxis(1,at=c(1,20,60,80),labels=c("-1kb","TSS","TES","1kb"))
		plot(t(kdapo_sub.tss[curgene.iv,]),ylim=c(ylimmin.tss,ylimmax.tss),type='l',main="TSS",lwd=3,ylab="",xaxt='n',xlab='',cex.main=1.5)
		lines(t(kdepo_sub.tss[curgene.iv,]),lty=2,col=unique(cols)[2],lwd=3)
		lines(t(kdgpo_sub.tss[curgene.iv,]),lty=2,col=unique(cols)[3],lwd=3)
		abline(v=60,col="gray",lwd=.4)
		axis(1,at=c(1,60,120),labels=c("-1kb","TSS","1kb"))
		plot(t(kdapo_sub.tes[curgene.iv,]),ylim=c(ylimmin.tes,ylimmax.tes),type='l',main="TES",lwd=3,ylab="",xaxt='n',xlab='',cex.main=1.5)
		lines(t(kdepo_sub.tes[curgene.iv,]),lty=2,col=unique(cols)[2],lwd=3)
		lines(t(kdgpo_sub.tes[curgene.iv,]),lty=2,col=unique(cols)[3],lwd=3)
		abline(v=60,col="gray",lwd=.4)
		axis(1,at=c(1,60,120),labels=c("-1kb","TES","1kb"))
		#legend("bottomright",legend=c("kdapo","kdepo","kdgpo"),col=unique(cols)[1:3],lwd=3,lty=c(1,2,2))
		mtext(sym,side=2,outer=T,line=1)
	}
dev.off()
```

[lineplots_selected_genes.pdf](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/lineplots_selected_genes.pdf)


## Looking into mutations in different samples, for lists of target genes

``` {r mutations, echo=TRUE, message=FALSE, results="hide", warning=FALSE, eval=FALSE}
goi<-rbind(trinuc,dement,mech)
goi$chrom<-genedata[match(goi[,2],genedata[,1]),"Chrom"]
goi$start<-genedata[match(goi[,2],genedata[,1]),"Start"]
goi$end<-genedata[match(goi[,2],genedata[,1]),"End"]
goi$strand<-genedata[match(goi[,2],genedata[,1]),"Strand"]

flipped<-as.character(goi$strand)
flipped[flipped=="+"]<-"neg"
flipped[flipped=="-"]<-"+"
flipped[flipped=="neg"]<-"-"
goi.gr<-GRanges(seqnames=goi$chrom,ranges=IRanges(start=goi$start,end=goi$end),strand=goi$strand)
goi.agr<-GRanges(seqnames=goi$chrom,ranges=IRanges(start=goi$start,end=goi$end),strand=flipped)
goi.exp.agr<-GRanges(seqnames=goi$chrom,ranges=IRanges(start=goi$start-3000,end=goi$end-3000),strand=flipped)

mm<-list()
amm<-list()
amm.exp<-list()
library(GenomicAlignments)
for(i in 1:length(targets[,1]))
{
	param<-ScanBamParam(tag="NM")
	cur<-readGAlignments(paste("/n/core/Bioinformatics/secondary/Workman/tas/",targets$Order[i],"/",targets$alias[i],"/accepted_hits.bam",sep=''),param=param)
	cur.gr<-GRanges(seqnames=seqnames(cur),ranges=IRanges(start=start(cur),end=end(cur)),strand=strand(cur),NM=elementMetadata(cur)$NM)

	#pull out reads for all the goi genes, on the right strand..
	#pairs <-findOverlapPairs(goi.gr,cur.gr)
	#mm[[i]]<-pairs@second$NM
	#apairs <-findOverlapPairs(goi.agr,cur.gr)
	#amm[[i]]<-apairs@second$NM
	apairs.exp <-findOverlapPairs(goi.exp.agr,cur.gr)
	amm.exp[[i]]<-apairs.exp@second$NM
}

png("total_mismatches.png",height=600,width=1000,pointsize=16)
par(mar=c(7,3,3,1))
barplot(unlist(lapply(amm,function(x){sum(x)/length(x)})),col=cols,names.arg=targets$alias,las=2,main="total mismatches / total reads")
dev.off()

png("gt2_mismatches.png",height=600,width=1000,pointsize=16)
par(mar=c(7,3,3,1))
barplot(unlist(lapply(amm,function(x){length(x[x >= 2])/length(x)})),col=cols,names.arg=targets$alias,las=2,main="percent of reads aligning to target genes with > 2 mismatches")
dev.off()

mmdata<-unlist(lapply(amm,function(x){length(x[x >= 2])/length(x)}))
cont_input<-mean(mmdata[c("s_cont1","s_cont2","s_cont3")])
cont_po<-mean(mmdata[c("s_contpo1","s_contpo2")])
kda_input<-mean(mmdata[c("s_kda1","s_kda2","s_kda4")])
kda_po<-mean(mmdata[c("s_kdapo1","s_kdapo2")])
kde_input<-mean(mmdata[c("s_kde1","s_kde2","s_kde3")])
kde_po<-mean(mmdata[c("s_kdepo1","s_kdepo2")])
kdg_input<-mean(mmdata[c("s_kdgin1","s_kdgin2")])
kdg_po<-mean(mmdata[c("s_kdgpo1","s_kdgpo2")])

to_plot<-cbind(c(cont_input,cont_po),c(kda_input,kda_po),c(kde_input,kde_po),c(kdg_input,kdg_po))
png("target_gene_mismatches.png",height=600,width=1000,pointsize=16)
barplot(to_plot,beside=T,names.arg=c("control","kda","kde","kdg"),col=c("grey","black"),ylab="% of reads aligning with > 2 mismatches",main="Alignment mismatch prevalance in target genes")
legend("topleft",legend=c("input","po"),pch=15,col=c("grey","black"))
dev.off()

to_plot_input<-c(cont_input,kda_input,kde_input,kdg_input)
to_plot_po<-c(cont_po,kda_po,kde_po,kdg_po)
png("target_gene_mismatches_sep.png",height=600,width=1200,pointsize=16)
par(mfrow=c(1,2))
barplot(to_plot_input,beside=T,names.arg=c("control","kda","kde","kdg"),col=c("grey"),ylab="% of reads aligning with > 2 mismatches",main="Alignment mismatch prevalance in target genes (input)")
barplot(to_plot_po,beside=T,names.arg=c("control","kda","kde","kdg"),col=c("black"),ylab="% of reads aligning with > 2 mismatches",main="polysome")
dev.off()

se_combined<-function(l)
{
	if(length(l)==2)
	{
		se<-1/2*sqrt(sd(l[[1]])/length(l[[1]]) + sd(l[[2]])/length(l[[2]]))
	}else if (length(l)==3)

	{
		se<-1/3*sqrt(sd(l[[1]])/length(l[[1]]) + sd(l[[2]])/length(l[[2]]) + sd(l[[3]])/length(l[[3]]))
	}
	se
}

mmdata<-unlist(lapply(amm.exp,function(x){length(x[x >= 2])/length(x)}))
names(mmdata)<-targets$alias
cont_input<-mean(mmdata[c("s_cont1","s_cont2","s_cont3")])
cont_po<-mean(mmdata[c("s_contpo1","s_contpo2")])
kda_input<-mean(mmdata[c("s_kda1","s_kda2","s_kda4")])
kda_po<-mean(mmdata[c("s_kdapo1","s_kdapo2")])
kde_input<-mean(mmdata[c("s_kde1","s_kde2","s_kde3")])
kde_po<-mean(mmdata[c("s_kdepo1","s_kdepo2")])
kdg_input<-mean(mmdata[c("s_kdgin1","s_kdgin2")])
kdg_po<-mean(mmdata[c("s_kdgpo1","s_kdgpo2")])
to_plot_input<-c(cont_input,kda_input,kde_input,kdg_input)
to_plot_po<-c(cont_po,kda_po,kde_po,kdg_po)

to_plot_input_se<-c(se_combined(amm.exp[1:3]),se_combined(amm.exp[4:6]),se_combined(amm.exp[13:15]),se_combined(amm.exp[38:39]))
to_plot_po_se<-c(se_combined(amm.exp[18:19]),se_combined(amm.exp[28:29]),se_combined(amm.exp[30:31]),se_combined(amm.exp[35:34]))
png("target_gene_mismatches_sep.exp3k.png",height=600,width=1200,pointsize=16)
par(mfrow=c(1,2))
bp<-barplot(to_plot_input,beside=T,names.arg=c("control","kda","kde","kdg"),col=c("lightgrey"),ylab="% of reads aligning with > 2 mismatches",main="Alignment mismatch prevalance in target genes (input)",ylim=c(0,.033))
arrows(x0=bp[,1],y0=to_plot_input-to_plot_input_se,x1=bp[,1],y1=to_plot_input+to_plot_input_se,angle=90,code=3,length=.04,lwd=1.5)
bp<-barplot(to_plot_po,beside=T,names.arg=c("control","kda","kde","kdg"),col=c("gray36"),ylab="% of reads aligning with > 2 mismatches",main="polysome",ylim=c(0,.025))
arrows(x0=bp[,1],y0=to_plot_po-to_plot_po_se,x1=bp[,1],y1=to_plot_po+to_plot_po_se,angle=90,code=3,length=.04,lwd=1.5)
dev.off()


```

If I take the list of all target genes and look for reads aligning to them, how many mismatches do those reads have in the various samples?

The first plot shows the total sum of mismatches divided by the number of reads aligning for each sample. The second shows the percent of reads that had 2 mismatches (the most that they could have).
```{r,out.width="600px"}
knitr::include_graphics("../data/total_mismatches.png")
knitr::include_graphics("../data/gt2_mismatches.png")
```

Selected samples:  

The third one is for each gene +/- 3k and the error bars represent the combined standard error of the mean.

```{r,out.width="600px"}
knitr::include_graphics("../data/target_gene_mismatches.png")
knitr::include_graphics("../data/target_gene_mismatches_sep.png")
knitr::include_graphics("../data/target_gene_mismatches_sep.exp3k.png")
```

### tas2/4/6 lineplots

``` {r tas2/4/6 gene lists lineplots, echo=TRUE, message=FALSE, results="hide", warning=FALSE, eval=FALSE}
trinuc<-read.table("trinuc_disease_genes.txt",sep='\t',header=F,as.is=T)
trinuc$ensid<-genedata[match(trinuc[,1],genedata[,2]),1]
genes<-trinuc[,1]
genes_ensid<-trinuc[,2]

	names(avgs)<-targets$alias
	mean_cont_10_11<-(avgs[["s_kdtas10po1"]][,2:81] + avgs[["s_kdtas10po2"]][,2:81] + avgs[["s_kdtas11po1"]][,2:81] + avgs[["s_kdtas11po2"]][,2:81])/4
	mean_kdtas2po<-(avgs[["s_kdtas2po1"]][,2:81] + avgs[["s_kdtas2po2"]][,2:81])/2
	kdtas2po_sub.gene <- mean_kdtas2po - mean_cont_10_11
	rownames(kdtas2po_sub.gene)<-avgs[["s_kdtas2po1"]][,1]

	names(tss_regions)<-targets$alias
	mean_cont_10_11<-(tss_regions[["s_kdtas10po1"]][,2:121] + tss_regions[["s_kdtas10po2"]][,2:121] + tss_regions[["s_kdtas11po1"]][,2:121] + tss_regions[["s_kdtas11po2"]][,2:121])/4
	mean_kdtas2po<-(tss_regions[["s_kdtas2po1"]][,2:121] + tss_regions[["s_kdtas2po2"]][,2:121])/2
	kdtas2po_sub.tss <- mean_kdtas2po - mean_cont_10_11
	rownames(kdtas2po_sub.tss)<-tss_regions[["s_kdtas2po1"]][,1]

	names(tes_regions)<-targets$alias
	mean_cont_10_11<-(tes_regions[["s_kdtas10po1"]][,2:121] + tes_regions[["s_kdtas10po2"]][,2:121] + tes_regions[["s_kdtas11po1"]][,2:121] + tes_regions[["s_kdtas11po2"]][,2:121])/2
	mean_kdtas2po<-(tes_regions[["s_kdtas2po1"]][,2:121] + tes_regions[["s_kdtas2po2"]][,2:121])/2
	kdtas2po_sub.tes <- mean_kdtas2po - mean_cont_10_11
	rownames(kdtas2po_sub.tes)<-tes_regions[["s_kdtas2po1"]][,1]

	names(avgs)<-targets$alias
	mean_cont_10_11<-(avgs[["s_kdtas10po1"]][,2:81] + avgs[["s_kdtas10po2"]][,2:81] + avgs[["s_kdtas11po1"]][,2:81] + avgs[["s_kdtas11po2"]][,2:81])/4
	mean_kdtas4po<-(avgs[["s_kdtas4po1"]][,2:81] + avgs[["s_kdtas4po2"]][,2:81])/2
	kdtas4po_sub.gene <- mean_kdtas4po - mean_cont_10_11
	rownames(kdtas4po_sub.gene)<-avgs[["s_kdtas4po1"]][,1]

	names(tss_regions)<-targets$alias
	mean_cont_10_11<-(tss_regions[["s_kdtas10po1"]][,2:121] + tss_regions[["s_kdtas10po2"]][,2:121] + tss_regions[["s_kdtas11po1"]][,2:121] + tss_regions[["s_kdtas11po2"]][,2:121])/4
	mean_kdtas4po<-(tss_regions[["s_kdtas4po1"]][,2:121] + tss_regions[["s_kdtas4po2"]][,2:121])/2
	kdtas4po_sub.tss <- mean_kdtas4po - mean_cont_10_11
	rownames(kdtas4po_sub.tss)<-tss_regions[["s_kdtas4po1"]][,1]

	names(tes_regions)<-targets$alias
	mean_cont_10_11<-(tes_regions[["s_kdtas10po1"]][,2:121] + tes_regions[["s_kdtas10po2"]][,2:121] + tes_regions[["s_kdtas11po1"]][,2:121] + tes_regions[["s_kdtas11po2"]][,2:121])/2
	mean_kdtas4po<-(tes_regions[["s_kdtas4po1"]][,2:121] + tes_regions[["s_kdtas4po2"]][,2:121])/2
	kdtas4po_sub.tes <- mean_kdtas4po - mean_cont_10_11
	rownames(kdtas4po_sub.tes)<-tes_regions[["s_kdtas4po1"]][,1]

	names(avgs)<-targets$alias
	mean_cont_10_11<-(avgs[["s_kdtas10po1"]][,2:81] + avgs[["s_kdtas10po2"]][,2:81] + avgs[["s_kdtas11po1"]][,2:81] + avgs[["s_kdtas11po2"]][,2:81])/4
	mean_kdtas6po<-(avgs[["s_kdtas6po1"]][,2:81] + avgs[["s_kdtas6po2"]][,2:81])/2
	kdtas6po_sub.gene <- mean_kdtas6po - mean_cont_10_11
	rownames(kdtas6po_sub.gene)<-avgs[["s_kdtas6po1"]][,1]

	names(tss_regions)<-targets$alias
	mean_cont_10_11<-(tss_regions[["s_kdtas10po1"]][,2:121] + tss_regions[["s_kdtas10po2"]][,2:121] + tss_regions[["s_kdtas11po1"]][,2:121] + tss_regions[["s_kdtas11po2"]][,2:121])/4
	mean_kdtas6po<-(tss_regions[["s_kdtas6po1"]][,2:121] + tss_regions[["s_kdtas6po2"]][,2:121])/2
	kdtas6po_sub.tss <- mean_kdtas6po - mean_cont_10_11
	rownames(kdtas6po_sub.tss)<-tss_regions[["s_kdtas6po1"]][,1]

	names(tes_regions)<-targets$alias
	mean_cont_10_11<-(tes_regions[["s_kdtas10po1"]][,2:121] + tes_regions[["s_kdtas10po2"]][,2:121] + tes_regions[["s_kdtas11po1"]][,2:121] + tes_regions[["s_kdtas11po2"]][,2:121])/2
	mean_kdtas6po<-(tes_regions[["s_kdtas6po1"]][,2:121] + tes_regions[["s_kdtas6po2"]][,2:121])/2
	kdtas6po_sub.tes <- mean_kdtas6po - mean_cont_10_11
	rownames(kdtas6po_sub.tes)<-tes_regions[["s_kdtas6po1"]][,1]

pdf("lineplots_selected_genes.tas246.pdf",height=8,width=24)
	goi.iv<-rownames(kdtas2po_sub.gene) %in% genes_ensid 
	for(i in 1:length(which(goi.iv)))
	{
		curgene.iv<-which(goi.iv)[i]
		sym<-ens2sym(rownames(kdtas2po_sub.gene)[curgene.iv])
		par(mfrow=c(1,3),oma=c(1,3,1,1))
		ylimmin.gene<-min(c(unlist(kdtas2po_sub.gene[curgene.iv,]),unlist(kdtas4po_sub.gene[curgene.iv,]),unlist(kdtas6po_sub.gene[curgene.iv,])))
		ylimmin.tss<-min(c(unlist(kdtas2po_sub.tss[curgene.iv,]),unlist(kdtas4po_sub.tss[curgene.iv,]),unlist(kdtas6po_sub.tss[curgene.iv,])))
		ylimmin.tes<-min(c(unlist(kdtas2po_sub.tes[curgene.iv,]),unlist(kdtas4po_sub.tes[curgene.iv,]),unlist(kdtas6po_sub.tes[curgene.iv,])))
		ylimmax.gene<-max(c(unlist(kdtas2po_sub.gene[curgene.iv,]),unlist(kdtas4po_sub.gene[curgene.iv,]),unlist(kdtas6po_sub.gene[curgene.iv,])))
		ylimmax.tss<-max(c(unlist(kdtas2po_sub.tss[curgene.iv,]),unlist(kdtas4po_sub.tss[curgene.iv,]),unlist(kdtas6po_sub.tss[curgene.iv,])))
		ylimmax.tes<-max(c(unlist(kdtas2po_sub.tes[curgene.iv,]),unlist(kdtas4po_sub.tes[curgene.iv,]),unlist(kdtas6po_sub.tes[curgene.iv,])))
		plot(t(kdtas2po_sub.gene[curgene.iv,]),ylim=c(ylimmin.gene,ylimmax.gene),type='l',main="Full Gene",lwd=3,ylab="average mut-wt signal",cex.main=1.5,cex.lab=1.5,xaxt='n',xlab='')
		lines(t(kdtas4po_sub.gene[curgene.iv,]),lty=2,col=unique(cols)[2],lwd=3)
		lines(t(kdtas6po_sub.gene[curgene.iv,]),lty=2,col=unique(cols)[3],lwd=3)
		abline(v=20,col="gray",lwd=.4)
		abline(v=60,col="gray",lwd=.4)
		axis(1,at=c(1,20,60,80),labels=c("-1kb","TSS","TES","1kb"))
		plot(t(kdtas2po_sub.tss[curgene.iv,]),ylim=c(ylimmin.tss,ylimmax.tss),type='l',main="TSS",lwd=3,ylab="",xaxt='n',xlab='',cex.main=1.5)
		lines(t(kdtas4po_sub.tss[curgene.iv,]),lty=2,col=unique(cols)[2],lwd=3)
		lines(t(kdtas6po_sub.tss[curgene.iv,]),lty=2,col=unique(cols)[3],lwd=3)
		abline(v=60,col="gray",lwd=.4)
		axis(1,at=c(1,60,120),labels=c("-1kb","TSS","1kb"))
		plot(t(kdtas2po_sub.tes[curgene.iv,]),ylim=c(ylimmin.tes,ylimmax.tes),type='l',main="TES",lwd=3,ylab="",xaxt='n',xlab='',cex.main=1.5)
		lines(t(kdtas4po_sub.tes[curgene.iv,]),lty=2,col=unique(cols)[2],lwd=3)
		lines(t(kdtas6po_sub.tes[curgene.iv,]),lty=2,col=unique(cols)[3],lwd=3)
		abline(v=60,col="gray",lwd=.4)
		axis(1,at=c(1,60,120),labels=c("-1kb","TES","1kb"))
		legend("bottomright",legend=c("kdtas2po","kdtas4po","kdtas6po"),col=unique(cols)[1:3],lwd=3,lty=c(1,2,2))
		mtext(sym,side=2,outer=T,line=1)
	}
dev.off()

pdf("lineplots_selected_genes.tas46.pdf",height=8,width=24)
	goi.iv<-rownames(kdtas4po_sub.gene) %in% genes_ensid 
	for(i in 1:length(which(goi.iv)))
	{
		curgene.iv<-which(goi.iv)[i]
		sym<-ens2sym(rownames(kdtas4po_sub.gene)[curgene.iv])
		par(mfrow=c(1,3),oma=c(1,3,1,1))
		ylimmin.gene<-min(c(unlist(kdtas4po_sub.gene[curgene.iv,]),unlist(kdtas6po_sub.gene[curgene.iv,])))
		ylimmin.tss<-min(c(unlist(kdtas4po_sub.tss[curgene.iv,]),unlist(kdtas6po_sub.tss[curgene.iv,])))
		ylimmin.tes<-min(c(unlist(kdtas4po_sub.tes[curgene.iv,]),unlist(kdtas6po_sub.tes[curgene.iv,])))
		ylimmax.gene<-max(c(unlist(kdtas4po_sub.gene[curgene.iv,]),unlist(kdtas6po_sub.gene[curgene.iv,])))
		ylimmax.tss<-max(c(unlist(kdtas4po_sub.tss[curgene.iv,]),unlist(kdtas6po_sub.tss[curgene.iv,])))
		ylimmax.tes<-max(c(unlist(kdtas4po_sub.tes[curgene.iv,]),unlist(kdtas6po_sub.tes[curgene.iv,])))
		plot(t(kdtas4po_sub.gene[curgene.iv,]),ylim=c(ylimmin.gene,ylimmax.gene),type='l',main="Full Gene",lwd=3,ylab="average mut-wt signal",cex.main=1.5,cex.lab=1.5,xaxt='n',xlab='')
		lines(t(kdtas6po_sub.gene[curgene.iv,]),lty=2,col=unique(cols)[2],lwd=3)
		abline(v=20,col="gray",lwd=.4)
		abline(v=60,col="gray",lwd=.4)
		axis(1,at=c(1,20,60,80),labels=c("-1kb","TSS","TES","1kb"))
		plot(t(kdtas4po_sub.tss[curgene.iv,]),ylim=c(ylimmin.tss,ylimmax.tss),type='l',main="TSS",lwd=3,ylab="",xaxt='n',xlab='',cex.main=1.5)
		lines(t(kdtas6po_sub.tss[curgene.iv,]),lty=2,col=unique(cols)[2],lwd=3)
		abline(v=60,col="gray",lwd=.4)
		axis(1,at=c(1,60,120),labels=c("-1kb","TSS","1kb"))
		plot(t(kdtas4po_sub.tes[curgene.iv,]),ylim=c(ylimmin.tes,ylimmax.tes),type='l',main="TES",lwd=3,ylab="",xaxt='n',xlab='',cex.main=1.5)
		lines(t(kdtas6po_sub.tes[curgene.iv,]),lty=2,col=unique(cols)[2],lwd=3)
		abline(v=60,col="gray",lwd=.4)
		axis(1,at=c(1,60,120),labels=c("-1kb","TES","1kb"))
		#legend("bottomright",legend=c("kdtas4po","kdtas6po"),col=unique(cols)[1:2],lwd=3,lty=c(1,2))
		mtext(sym,side=2,outer=T,line=1)
	}
dev.off()
```

[lineplots_selected_genes.tas246.pdf](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/lineplots_selected_genes.pdf)

## heatmap

``` {r exp heatmap, echo=TRUE, message=FALSE, results="hide", warning=FALSE, eval=FALSE}
goi<-c("XDH","MOCS2","MBIP","ZZZ3","GNAI3","GNAI2","GNB1","GNB4","FMR1","HTT","JPH3","AFF2","SOD1","HNRNPA1","ATXN10","ATXN2","EIF5A2","DBN1","PRKN","UBQLN2")
goi_ensid<-genedata[match(goi,genedata[,2]),1]
temp<-log2normCounts[rownames(log2normCounts) %in% goi_ensid,c("s_kdtas2po1","s_kdtas2po2","s_kdtas4po1","s_kdtas4po2","s_kdtas6po1","s_kdtas6po2","s_kdtas10po1","s_kdtas10po2","s_kdtas11po1","s_kdtas11po2")]
rownames(temp)<-paste(rownames(temp)," (",genedata[match(rownames(temp),genedata[,1]),2],")",sep='')
finite.iv<-apply(temp,2,is.finite)
temp[!finite.iv]<- -3
h<-heatmap(temp,Colv=NA)
#pdf("heatmap_exp_tas.pdf")
png("heatmap_exp_tas.png",height=800,width=900,pointsize=16)
source("/n/projects/mcm/R/utilities/rownorm.R")
source("/n/projects/mcm/R/utilities/mcm_image.R")
temp<-row.norm(temp,divSD=T)
temp[temp < -3] <- -3
temp[temp > 3 ] <- 3
nf<-layout(matrix(data=c(1,1,1,2,0,0),nrow=3,ncol=2),widths=c(13,3),TRUE)
mcm_image(temp[rev(h$rowInd),h$colInd],colors=colorRampPalette(brewer.pal(9,"Reds"))(100),zlim=c(-3,3),cexRow=.7,lab=5,title="log2(normCounts), row normalized",cexCol=1)
dev.off()
```

```{r,out.width="600px"}
knitr::include_graphics("../data/heatmap_exp_tas.png")
```

[heatmap_exp_tas.pdf](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/heatmap_exp_tas.pdf)

``` {r exp heatmap avgd, echo=TRUE, message=FALSE, results="hide", warning=FALSE, eval=FALSE}
goi<-c("XDH","MOCS2","MBIP","ZZZ3","GNAI3","GNAI2","GNB1","GNB4","FMR1","HTT","JPH3","AFF2","SOD1","HNRNPA1","ATXN10","ATXN2","EIF5A2","DBN1","PRKN","UBQLN2")
goi_ensid<-genedata[match(goi,genedata[,2]),1]
temp<-log2normCounts[rownames(log2normCounts) %in% goi_ensid,c("s_kdtas2po1","s_kdtas2po2","s_kdtas4po1","s_kdtas4po2","s_kdtas6po1","s_kdtas6po2","s_kdtas10po1","s_kdtas10po2","s_kdtas11po1","s_kdtas11po2")]
temp<-edger.rpkm[rownames(edger.rpkm) %in% goi_ensid,c("rpkm.s_kdtas2po1","rpkm.s_kdtas2po2","rpkm.s_kdtas4po1","rpkm.s_kdtas4po2","rpkm.s_kdtas6po1","rpkm.s_kdtas6po2","rpkm.s_kdtas10po1","rpkm.s_kdtas10po2","rpkm.s_kdtas11po1","rpkm.s_kdtas11po2")]
rownames(temp)<-paste(rownames(temp)," (",genedata[match(rownames(temp),genedata[,1]),2],")",sep='')
temp2<-data.frame(tas2=rowMeans(temp[,1:2],na.rm=T),tas4=rowMeans(temp[,3:4],na.rm=T),tas6=rowMeans(temp[,5:6],na.rm=T),tas10_11=rowMeans(temp[,7:10],na.rm=T))
finite.iv<-apply(temp2,2,is.finite)
temp2[!finite.iv]<- -3
h<-heatmap(as.matrix(temp2),Colv=NA)
#pdf("heatmap_exp_tas.avg.rpkm.pdf")
png("heatmap_exp_tas.avg.rpkm.png",height=800,width=900,pointsize=16)
source("/n/projects/mcm/R/utilities/rownorm.R")
source("/n/projects/mcm/R/utilities/mcm_image.R")
temp2<-row.norm(temp2,divSD=T)
temp2[temp2 < -3] <- -3
temp2[temp2 > 3 ] <- 3
nf<-layout(matrix(data=c(1,1,1,2,0,0),nrow=3,ncol=2),widths=c(13,3),TRUE)
mcm_image(temp2[rev(h$rowInd),h$colInd],colors=colorRampPalette(brewer.pal(9,"Reds"))(100),zlim=c(-3,3),cexRow=.7,lab=5,title="RPKM, row normalized",cexCol=1)
dev.off()
```

```{r,out.width="600px"}
knitr::include_graphics("../data/heatmap_exp_tas.avg.png")
knitr::include_graphics("../data/heatmap_exp_tas.avg.rpkm.png")
```

[heatmap_exp_tas.avg.pdf](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/heatmap_exp_tas.avg.pdf)  
[heatmap_exp_tas.avg.rpkm.pdf](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/heatmap_exp_tas.avg.rpkm.pdf)  

## DE genes on X or Y chromosomes

``` {r de genes x y, echo=TRUE, message=FALSE, results="hide", warning=FALSE, eval=FALSE}

selected_conts<-c("kda_vs_cont", "kde_vs_cont","kdgin_vs_cont","kdapo_vs_contpo", "kdepo_vs_contpo", "kdgpo_vs_contpo","kdtas2po_vs_kdtas10_11_po", "kdtas4po_vs_kdtas10_11_po", "kdtas6po_vs_kdtas10_11_po")
k<-1
d<-list()
f<-list()
for(i in 1:length(selected_conts))
{
	for(j in c("up","dn"))
	{
		d[[k]]<-read.table(paste("../results/DE/",selected_conts[i],".de.",j,".xls",sep=''),sep='\t',as.is=T,header=T,quote="",comment.char="")
		chrx_y.iv<-d[[k]]$Chrom %in% c("chrX","chrY")
		f[[k]]<-d[[k]][chrx_y.iv,]
		if(length(f[[k]][,1]) > 0)
		{
			write.table(f[[k]],paste("../results/DE_xy/",selected_conts[i],".de.",j,".chrX_chrY.xls",sep=''),sep='\t',quote=F,row.names=F)
		}
		k<-k+1
	}
}
xy.df<-data.frame(contrast=selected_conts,up.dn=c(rep("up",9),rep("dn",9)),de=unlist(lapply(f,function(x){length(x[,1])})))
write.table(xy.df,"xy.df.txt",sep='\t',quote=F,row.names=F)

```

There were a total of 203 genes in these contrasts that had differential expression and were on chrX or chrY. I created a file with these genes for each contrast in this folder:

[DE_xy](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/results/DE_xy)

``` {r xy table, echo=TRUE, message=FALSE, warning=FALSE, eval=TRUE}
xy.df<-read.table("xy.df.txt",sep='\t',as.is=T,header=T)
kable(xy.df,caption="num X or Y de genes")
```

## Revamping some figures

Here is the 3-way venn diagram without tas2.

```{r,out.width="600px"}
	knitr::include_graphics("../data/2venn.yellow_venns.png")
#Here is a file with genes in tas4 only:  [2venn.yellowvenn.10.2465.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/2venn.yellowvenn.10.2465.txt)  genes in both: 2venn.yellowvenn.11.84.txt 2venn.yellowvenn.01.262.txt
```


Here is a set of line plots without tas2.  

[lineplots_selected_genes.tas46.pdf](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/lineplots_selected_genes.tas46.pdf)


## Files for venn diagrams with up/down

[3venn.yellowvenn.100.updown.46.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.100.updown.46.txt)  
[3venn.yellowvenn.001.updown.247.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.001.updown.247.txt)  
[3venn.yellowvenn.111.updown.10.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.111.updown.10.txt)  
[3venn.yellowvenn.010.updown.2447.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.010.updown.2447.txt)  
[3venn.yellowvenn.110.updown.18.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.110.updown.18.txt)  
[3venn.yellowvenn.101.updown.15.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.101.updown.15.txt)  
[3venn.yellowvenn.011.updown.74.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.011.updown.74.txt)  

[3venn.kdapo.kdgpo.kdepo.100.updown.127.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.kdapo.kdgpo.kdepo.100.updown.127.txt)  
[3venn.kdapo.kdgpo.kdepo.010.updown.194.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.kdapo.kdgpo.kdepo.010.updown.194.txt)  
[3venn.kdapo.kdgpo.kdepo.001.updown.254.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.kdapo.kdgpo.kdepo.001.updown.254.txt)  
[3venn.kdapo.kdgpo.kdepo.110.updown.20.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.kdapo.kdgpo.kdepo.110.updown.20.txt)  
[3venn.kdapo.kdgpo.kdepo.011.updown.152.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.kdapo.kdgpo.kdepo.011.updown.152.txt)  
[3venn.kdapo.kdgpo.kdepo.101.updown.124.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.kdapo.kdgpo.kdepo.101.updown.124.txt)  
[3venn.kdapo.kdgpo.kdepo.111.updown.173.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.kdapo.kdgpo.kdepo.111.updown.173.txt)  

## Files split into up and down or both

[3venn.yellowvenn.100.updown.46.34.up.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.100.updown.46.34.up.txt)  
[3venn.yellowvenn.100.updown.46.12.dn.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.100.updown.46.12.dn.txt)  
[3venn.yellowvenn.001.updown.247.180.up.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.001.updown.247.180.up.txt)  
[3venn.yellowvenn.001.updown.247.67.dn.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.001.updown.247.67.dn.txt)  
[3venn.yellowvenn.111.updown.10.6.up.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.111.updown.10.6.up.txt)  
[3venn.yellowvenn.111.updown.10.4.dn.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.111.updown.10.4.dn.txt)  
[3venn.yellowvenn.010.updown.2447.2315.up.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.010.updown.2447.2315.up.txt)  
[3venn.yellowvenn.010.updown.2447.132.dn.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.010.updown.2447.132.dn.txt)  
[3venn.yellowvenn.110.updown.18.11.up.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.110.updown.18.11.up.txt)  
[3venn.yellowvenn.110.updown.18.7.dn.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.110.updown.18.7.dn.txt)  
[3venn.yellowvenn.101.updown.15.8.up.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.101.updown.15.8.up.txt)  
[3venn.yellowvenn.101.updown.15.7.dn.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.101.updown.15.7.dn.txt)  
[3venn.yellowvenn.011.updown.74.4.both.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.011.updown.74.4.both.txt)  
[3venn.yellowvenn.011.updown.74.54.up.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.011.updown.74.54.up.txt)  
[3venn.yellowvenn.011.updown.74.16.dn.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.yellowvenn.011.updown.74.16.dn.txt)  
[3venn.kdapo.kdgpo.kdepo.100.updown.127.36.up.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.kdapo.kdgpo.kdepo.100.updown.127.36.up.txt)  
[3venn.kdapo.kdgpo.kdepo.100.updown.127.91.dn.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.kdapo.kdgpo.kdepo.100.updown.127.91.dn.txt)  
[3venn.kdapo.kdgpo.kdepo.010.updown.194.42.up.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.kdapo.kdgpo.kdepo.010.updown.194.42.up.txt)  
[3venn.kdapo.kdgpo.kdepo.010.updown.194.152.dn.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.kdapo.kdgpo.kdepo.010.updown.194.152.dn.txt)  
[3venn.kdapo.kdgpo.kdepo.001.updown.254.24.up.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.kdapo.kdgpo.kdepo.001.updown.254.24.up.txt)  
[3venn.kdapo.kdgpo.kdepo.001.updown.254.230.dn.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.kdapo.kdgpo.kdepo.001.updown.254.230.dn.txt)  
[3venn.kdapo.kdgpo.kdepo.110.updown.20.7.up.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.kdapo.kdgpo.kdepo.110.updown.20.7.up.txt)  
[3venn.kdapo.kdgpo.kdepo.110.updown.20.13.dn.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.kdapo.kdgpo.kdepo.110.updown.20.13.dn.txt)  
[3venn.kdapo.kdgpo.kdepo.011.updown.152.14.up.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.kdapo.kdgpo.kdepo.011.updown.152.14.up.txt)  
[3venn.kdapo.kdgpo.kdepo.011.updown.152.138.dn.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.kdapo.kdgpo.kdepo.011.updown.152.138.dn.txt)  
[3venn.kdapo.kdgpo.kdepo.101.updown.124.29.up.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.kdapo.kdgpo.kdepo.101.updown.124.29.up.txt)  
[3venn.kdapo.kdgpo.kdepo.101.updown.124.95.dn.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.kdapo.kdgpo.kdepo.101.updown.124.95.dn.txt)  
[3venn.kdapo.kdgpo.kdepo.111.updown.173.14.up.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.kdapo.kdgpo.kdepo.111.updown.173.14.up.txt)  
[3venn.kdapo.kdgpo.kdepo.111.updown.173.159.dn.txt](https://webfs/n/facility/Bioinformatics/analysis/Workman/tas/cbio.tas.103/data/3venn.kdapo.kdgpo.kdepo.111.updown.173.159.dn.txt)  

